---
title: 'Lipidomics analysis'
author: "Fanning Xia"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Purpose

To characterize and compare protein and lipid profiles from surface fish and 2 cavefish 2-cell stage embryos to see if there's any deposition difference as maternal nutrients.

# 0. Preparation

```{r}
# read the file from Compound Discover output
# filtering information see the PAR in proteomics drive 
# /n/proteomics/rohner/fx2482/PROT-922/Rohner-Xia_PROT-922-MAR_2023-12-14

# need to filter out the lipids that have average(sysH2O) or average(X-Bf) > 5 * min(ave(SF, PA and MO))


# set working directory
setwd("/Volumes/projects/fx2482/lipidomics-2cellstage/")

library(ggplot2)
library(NormalyzerDE)
library(preprocessCore)
library(dplyr)
library(limma)
library(tidyr)
library(ggfortify)
library(cluster)
library(ggsci)
library(devtools)
#install_github("metabolomicsworkbench/RefMet")
library(RefMet)
library(ggrepel)
library(tidyverse)
library(scales)
#show_col(pal_npg("nrc")(9))
library(gridExtra)
library(vangogh)
library(ggpubr)
library(ggExtra)
library(car)
library(FSA)
library(reshape2)


pvg35 <- c(vangogh_palette("StarryRhone"),
           vangogh_palette("SelfPortrait"),
           vangogh_palette("Irises"),
           vangogh_palette("CafeDeNuit"),
           vangogh_palette("Landscape"),
           vangogh_palette("Shoes"),
           vangogh_palette("SunflowersLondon"))
pvg20 <- c(vangogh_palette("Landscape"),
           vangogh_palette("Shoes"),
           vangogh_palette("CafeTerrace"), 
           vangogh_palette("Chaise"))
pvg15 <- pvg20[c(1:8,10,13:18)]
pvg3 <- c("#979FA2","#EE9B00","#93D2DF")
```


# 1. merge different rows (manually) and filter out some low quality lipid species

## 1.1 filter out low quality lipid species

#### positive ion mode

```{r}

# positive ion mode first

pos <- read.csv("input/individual/need_filtering_Am-Embryo_pos_6_compounds.csv")

pos$MO <- rowMeans(pos[, grep("^Group.Area..M", colnames(pos), value = TRUE)], na.rm = TRUE)
pos$PA <- rowMeans(pos[, grep("^Group.Area..P", colnames(pos), value = TRUE)], na.rm = TRUE)
pos$SF <- rowMeans(pos[, grep("^Group.Area..S\\d+", colnames(pos), value = TRUE)], na.rm = TRUE)


pos$SysH2O <- rowMeans(pos[, grep("^Group.Area..SysH2O", colnames(pos), value = TRUE)], na.rm = TRUE)
pos$Bf <- rowMeans(pos[, grep("^Group.Area..X.Bf", colnames(pos), value = TRUE)], na.rm = TRUE)

# keep every row if both mean(sysH2O) and mean(X-Bf) are lower than 5* min(mean(SF), mean(PA) and mean(MO))

pos_filtered <- pos[pos$SysH2O < 5*pos$MO &
                      pos$SysH2O < 5*pos$PA &
                      pos$SysH2O < 5*pos$SF &
                      pos$Bf < 5*pos$MO &
                      pos$Bf < 5*pos$PA &
                      pos$Bf < 5*pos$SF, ]


#write.csv(pos_filtered, "input/individual/negative control filtered_Am-Embryo_pos_6_compounds.csv")

rm(pos)
```

#### negative ion mode

```{r}
# negative ion mode then

neg <- read.csv("input/individual/need_filtering_Am-Embryo_neg_3_compounds.csv")

neg$MO <- rowMeans(neg[, grep("^Group.Area..M", colnames(neg), value = TRUE)], na.rm = TRUE)
neg$PA <- rowMeans(neg[, grep("^Group.Area..P", colnames(neg), value = TRUE)], na.rm = TRUE)
neg$SF <- rowMeans(neg[, grep("^Group.Area..S\\d+", colnames(neg), value = TRUE)], na.rm = TRUE)


neg$SysH2O <- rowMeans(neg[, grep("^Group.Area..SysH2O", colnames(neg), value = TRUE)], na.rm = TRUE)
neg$Bf <- rowMeans(neg[, grep("^Group.Area..X.Bf", colnames(neg), value = TRUE)], na.rm = TRUE)

# keep every row if both mean(sysH2O) and mean(X-Bf) are lower than 5* min(mean(SF), mean(PA) and mean(MO))

neg_filtered <- neg[neg$SysH2O < 5*neg$MO &
                      neg$SysH2O < 5*neg$PA &
                      neg$SysH2O < 5*neg$SF &
                      neg$Bf < 5*neg$MO &
                      neg$Bf < 5*neg$PA &
                      neg$Bf < 5*neg$SF, ]

#write.csv(neg_filtered, "input/individual/negative control filtered_Am-Embryo_neg_3_compounds.csv")

rm(neg)
```

## 1.2 merge the repetitive rows

```{r}

library(dplyr)
merged.pos <- pos_filtered %>%
  group_by(Name) %>%
  summarise(across(.cols = starts_with("Group.Area.."), .fns = sum))

merged.neg <- neg_filtered %>%
  group_by(Name) %>%
  summarise(across(.cols = starts_with("Group.Area.."), .fns = sum))

head(merged.pos)
head(merged.neg)

# check if row number is correct
length(unique(pos_filtered$Name))
length(unique(neg_filtered$Name))

rm(pos_filtered, neg_filtered)
```

## 1.3 check for pre-normalization matrix

#### used to decide which sample to keep

positive ion mode

```{r}
cols<-rep(c("darkcyan","darkorange3"),each=3)
boxplot(log2(merged.pos[,2:31]),col=cols,main="before norm in positive ion mode")
colSums(merged.pos[,2:31])
```

```{r}
cor_plot <- pheatmap::pheatmap(cor(log2(merged.pos[,2:31])), main = "before norm in positive ion mode")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/before-norm-correlation-plot.png", plot = cor_plot,width = 10, height = 10, units = "in")
```

negative ion mode

```{r}
boxplot(log2(merged.neg[,2:24]),col=cols,main="before norm in negative ion mode")
colSums(merged.neg[,2:24])
```

```{r}
cor_plot <- pheatmap::pheatmap(cor(log2(merged.neg[,2:24])), main = "before norm in negative ion mode")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/before-norm-correlation-plot.png", plot = cor_plot,width = 10, height = 10, units = "in")

rm(cor_plot)
```

decided to remove: M5 (N), P4(P), S3(P) and S1(N+P). Both positive and negative ion mode will be removed.

S3 positive ion mode cluster differently than all other surface replicates/samples. I've tried it before. S3 is still the outlier if we include it.

#### remove outliers and technical negative controls (water & buffer)

```{r}
merged.pos.filtered <- merged.pos[,c(1:5, 7:10, 12:13,15,17:19)]
boxplot(log2(merged.pos.filtered[,2:15]),col=cols,main="before norm in positive ion mode")
pheatmap::pheatmap(cor(log2(merged.pos.filtered[,2:15])), main = "before norm in positive ion mode")


merged.neg.filtered <- merged.neg[,c(1:9, 11:12,14,16:18)]
boxplot(log2(merged.neg.filtered[,2:15]),col=cols,main="before norm in negative ion mode")
pheatmap::pheatmap(cor(log2(merged.neg.filtered[,2:15])), main = "before norm in negative ion mode")

```

# 2. Normalization methods

## 2.1 load (and create) data and sampleinfo for heatmap

```{r}
#write.table(merged.pos.filtered[,2:15], file='input/individual/prep_normDE_outliersremoved_Am-Embryo_pos_6_compounds.tsv', quote=FALSE, sep='\t', col.names = NA)
#write.table(merged.neg.filtered[,2:15], file='input/individual/prep_normDE_outliersremoved_Am-Embryo_neg_3_compounds.tsv', quote=FALSE, sep='\t', col.names = NA)
```

```{r}
# create a sample info table

pos.filtered.sampleinfo <- data.frame(
  sample = colnames(merged.pos.filtered[,2:15]),
  Population = c("Molino","Molino","Molino","Molino","Molino","Pachón","Pachón","Pachón","Pachón","Pachón","Surface_fish","Surface_fish","Surface_fish","Surface_fish"),
  stringsAsFactors = FALSE  # This ensures that text columns are not converted to factors
)

pos.filtered.sampleinfo$Population <- factor(pos.filtered.sampleinfo$Population, levels = c("Surface_fish", "Pachón", "Molino"))

#write.table(pos.filtered.sampleinfo, file='input/sampleinfo/prep_normDE_outliersremoved_pos.sampleinfo.tsv', quote=FALSE, sep='\t', col.names = NA)

neg.filtered.sampleinfo <- data.frame(
  sample = colnames(merged.neg.filtered[,2:15]),
  Population = c("Molino","Molino","Molino","Molino","Molino","Pachón","Pachón","Pachón","Pachón","Pachón","Surface_fish","Surface_fish","Surface_fish","Surface_fish"),
  stringsAsFactors = FALSE  # This ensures that text columns are not converted to factors
)

neg.filtered.sampleinfo$Population <- factor(neg.filtered.sampleinfo$Population, levels = c("Surface_fish", "Pachón", "Molino"))

#write.table(neg.filtered.sampleinfo, file='input/sampleinfo/prep_normDE_outliersremoved_neg.sampleinfo.tsv', quote=FALSE, sep='\t', col.names = NA)
```

## 2.2 NormalyzerDE evaluation

### positive ion mode

Removed outliers group

```{r}
jobName <- "vignette_run_outliersremoved_pos"
dataFp <- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/individual/prep_normDE_outliersremoved_Am-Embryo_pos_6_compounds.tsv"
designFp<- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/sampleinfo/prep_normDE_outliersremoved_pos.sampleinfo.tsv"

experimentObj <- setupRawDataObject(dataFp, designFp, "default", TRUE, "sample", "Population")
normObj <- getVerifiedNormalyzerObject(jobName, experimentObj)

# 2. generate normalizations
normResults <- normMethods(normObj)

# 3. generate performance measures
normResultsWithEval <- analyzeNormalizations(normResults)

# 4. output matrices to file
jobDir <- setupJobDir("vignette_run_outliersremoved_pos", "/Volumes/projects/fx2482/lipidomics-2cellstage/output/")
#writeNormalizedDatasets(normResultsWithEval, jobDir)

# 5. generate evaluation plots
#generatePlots(normResultsWithEval, jobDir)

rm(jobDir,jobName, dataFp,designFp,experimentObj,normObj,normResults,normResultsWithEval)
```

### negative ion mode

Removed outliers group

```{r}
jobName <- "vignette_run_outliersremoved_neg"
dataFp <- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/individual/prep_normDE_outliersremoved_Am-Embryo_neg_3_compounds.tsv"
designFp<- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/sampleinfo/prep_normDE_outliersremoved_neg.sampleinfo.tsv"

experimentObj <- setupRawDataObject(dataFp, designFp, "default", TRUE, "sample", "Population")
normObj <- getVerifiedNormalyzerObject(jobName, experimentObj)

# 2. generate normalizations
normResults <- normMethods(normObj)

# 3. generate performance measures
normResultsWithEval <- analyzeNormalizations(normResults)

# 4. output matrices to file
jobDir <- setupJobDir("vignette_run_outliersremoved_neg", "/Volumes/projects/fx2482/lipidomics-2cellstage/output/")
#writeNormalizedDatasets(normResultsWithEval, jobDir)

# 5. generate evaluation plots
#generatePlots(normResultsWithEval, jobDir)

rm(jobDir,jobName, dataFp,designFp,experimentObj,normObj,normResults,normResultsWithEval)
```


# 3. Differential expression analysis

### positive ion mode - median normalization

```{r}
# 1. Setup folders and data matrices
jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_pos/"
designFp<- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/sampleinfo/prep_normDE_outliersremoved_pos.sampleinfo.tsv"

bestNormMatPath <- paste(jobDir, "median-normalized.txt", sep="/")
experimentObj <- setupRawContrastObject(bestNormMatPath, designFp, "sample")
nst <- NormalyzerStatistics(experimentObj, logTrans=FALSE)

# 2. Calculate statistics
# Now we are ready to perform the contrasts. Contrasts are provided as a vector in the format c("condA-condB", "condB-condC"), where condX is the group levels.

comparisons <- c("Molino-Surface_fish", "Pachón-Surface_fish")
nst <- calculateContrasts(nst, comparisons, condCol="Population", leastRepCount=2)

# 3. Generate final matrix and output
pos.Df <- generateAnnotatedMatrix(nst)
pos.jobDir <- setupJobDir("median-norm/DEG_analysis", "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_pos/")

# load lipid names in the dataframe
test <- merged.pos.filtered
test$V1 <- as.integer(rownames(merged.pos.filtered))
pos.annotDf <- left_join(test[,c(1,16)], pos.Df,by = c("V1" = "NA") )
pos.annotDf <- pos.annotDf[,c(1,3:23)]

#write.csv(pos.annotDf, file=paste(pos.jobDir, "stat_table.csv", sep="/"))
#generateStatsReport(nst, "Vignette stats", pos.jobDir)

rm(jobDir,designFp,bestNormMatPath,experimentObj,nst,comparisons,test,pos.Df)
```


### negative ion mode - GI normalization

```{r}
# 1. Setup folders and data matrices
#jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/"
#designFp<- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/sampleinfo/prep_normDE_outliersremoved_neg.sampleinfo.tsv"

#bestNormMatPath <- paste(jobDir, "GI-normalized.txt", sep="/")
#experimentObj <- setupRawContrastObject(bestNormMatPath, designFp, "sample")
#nst <- NormalyzerStatistics(experimentObj, logTrans=FALSE)

# 2. Calculate statistics
# Now we are ready to perform the contrasts. Contrasts are provided as a vector in the format c("condA-condB", "condB-condC"), where condX is the group levels.

#comparisons <- c("Molino-Surface_fish", "Pachón-Surface_fish")
#nst <- calculateContrasts(nst, comparisons, condCol="group", leastRepCount=2)

# 3. Generate final matrix and output
#neg.Df <- generateAnnotatedMatrix(nst)
#neg.jobDir <- setupJobDir("GI-norm/DEG_analysis", "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/")

# load lipid names in the dataframe
#test <- merged.neg.filtered
#test$V1 <- as.integer(rownames(merged.neg.filtered))
#neg.annotDf <- left_join(test[,c(1,16)], neg.Df,by = c("V1" = "NA") )
#neg.annotDf <- neg.annotDf[,c(1,3:23)]

#write.csv(neg.annotDf, file=paste(neg.jobDir, "stat_table.csv", sep="/"))
#generateStatsReport(nst, "Vignette stats", neg.jobDir)

#rm(jobDir,designFp,bestNormMatPath,experimentObj,nst,comparisons,test,neg.Df)
```


### negative ion mode - median normalization

```{r}
# 1. Setup folders and data matrices
jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/"
designFp<- "/Volumes/projects/fx2482/lipidomics-2cellstage/input/sampleinfo/prep_normDE_outliersremoved_neg.sampleinfo.tsv"

bestNormMatPath <- paste(jobDir, "median-normalized.txt", sep="/")
experimentObj <- setupRawContrastObject(bestNormMatPath, designFp, "sample")
nst <- NormalyzerStatistics(experimentObj, logTrans=FALSE)

# 2. Calculate statistics
# Now we are ready to perform the contrasts. Contrasts are provided as a vector in the format c("condA-condB", "condB-condC"), where condX is the group levels.

comparisons <- c("Molino-Surface_fish", "Pachón-Surface_fish")
nst <- calculateContrasts(nst, comparisons, condCol="Population", leastRepCount=2)

# 3. Generate final matrix and output
neg.Df <- generateAnnotatedMatrix(nst)
neg.jobDir <- setupJobDir("median-norm/DEG_analysis", "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/")

# load lipid names in the dataframe
test <- merged.neg.filtered
test$V1 <- as.integer(rownames(merged.neg.filtered))
neg.annotDf <- left_join(test[,c(1,16)], neg.Df,by = c("V1" = "NA") )
neg.annotDf <- neg.annotDf[,c(1,3:23)]

#write.csv(neg.annotDf, file=paste(neg.jobDir, "stat_table.csv", sep="/"))
#generateStatsReport(nst, "Vignette stats", neg.jobDir)

rm(jobDir,designFp,bestNormMatPath,experimentObj,nst,comparisons,test,neg.Df)
```


## 3.1 check if two collection methods are well separated

positive ion mode

```{r}

# Comparison as a whole

# comparison data creation

df <- merged.pos.filtered[,c(1,7:11)]


# Reshaping data to long format
long_df <- pivot_longer(df, cols = -Name, names_to = "Sample", values_to = "Value")
long_df <- long_df %>%
  mutate(Method = case_when(
    Sample %in% c("Group.Area..P1_pos", "Group.Area..P2_pos") ~ "NS", 
    Sample %in% c("Group.Area..P3_pos", "Group.Area..P5_pos", "Group.Area..P6_pos") ~ "IVF"
  ))

# Check normality with visual methods
ggplot(long_df, aes(sample = Value)) +
  facet_wrap(~ Method) +
  stat_qq() +
  stat_qq_line() +
  theme_minimal() +
  labs(title = "Q-Q Plots by Method") # They are not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Method, data = long_df) # homogeneous

# Perform Mann-Whitney U test
wilcox.test(Value ~ Method, data = long_df) # non-significant

rm(df,long_df)
```

```{r}
# Comparison individually

df <- merged.pos.filtered[,c(1,7:11)]

# Create an empty vector to store p-values
p_values <- numeric(nrow(df))

# Loop through each row/gene
for (i in 1:nrow(df)) {
  # Subset data for the current row/gene
  data_row <- df[i, ]
  
  # Subset data based on the method of collection
  data_method_ns <- data_row[colnames(data_row) %in% c("Group.Area..P1_pos", "Group.Area..P2_pos")]
  data_method_ivf <- data_row[colnames(data_row) %in% c("Group.Area..P3_pos", "Group.Area..P5_pos", "Group.Area..P6_pos")]
  
  # Perform t-test
  t_test_result <- t.test(data_method_ns, data_method_ivf)
  
  # Extract p-value and store it
  p_values[i] <- t_test_result$p.value
}

df$p.value <- p_values
df$p.adj <- p.adjust(p_values, method = "BH")

# print any significant expressed lipid names
print(df[df$p.adj < 0.05, ])

rm(df,p_values,i,data_row,data_method_ivf,data_method_ns,t_test_result)

```

negative ion mode

```{r}

# Comparison as a whole

# comparison data creation

df <- merged.neg.filtered[,c(1,7:11)]

# Reshaping data to long format
long_df <- pivot_longer(df, cols = -Name, names_to = "Sample", values_to = "Value")
long_df <- long_df %>%
  mutate(Method = case_when(
    Sample %in% c("Group.Area..P1_neg", "Group.Area..P2_neg") ~ "NS", 
    Sample %in% c("Group.Area..P3_neg", "Group.Area..P5_neg", "Group.Area..P6_neg") ~ "IVF"
  ))

# Check normality with visual methods
ggplot(long_df, aes(sample = Value)) +
  facet_wrap(~ Method) +
  stat_qq() +
  stat_qq_line() +
  theme_minimal() +
  labs(title = "Q-Q Plots by Method") # They are not normally distributed

by(long_df$Value, long_df$Method, shapiro.test) # not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Method, data = long_df) # homogeneous

# Perform Mann-Whitney U test
wilcox.test(Value ~ Method, data = long_df) # significant

rm(df,long_df)
```

```{r}
# Comparison individually

df <- merged.neg.filtered[,c(1,7:11)]

# Create an empty vector to store p-values
p_values <- numeric(nrow(df))

# Loop through each row/gene
for (i in 1:nrow(df)) {
  # Subset data for the current row/gene
  data_row <- df[i, ]
  
  # Subset data based on the method of collection
  data_method_ns <- data_row[colnames(data_row) %in% c("Group.Area..P1_neg", "Group.Area..P2_neg")]
  data_method_ivf <- data_row[colnames(data_row) %in% c("Group.Area..P3_neg", "Group.Area..P5_neg", "Group.Area..P6_neg")]
  
  # Perform t-test
  t_test_result <- t.test(data_method_ns, data_method_ivf)
  
  # Extract p-value and store it
  p_values[i] <- t_test_result$p.value
}

df$p.value <- p_values
df$p.adj <- p.adjust(df$p.value, method = "BH")

# print any significant expressed lipid names
print(df[df$p.adj < 0.05, ])

rm(i,df,p_values,data_row,data_method_ns,data_method_ivf,t_test_result)
```

No individual lipid species were found in negative ion mode. I will now assume it has not changed to the specific lipid level.

## 3.2 check for post-normalization matrix

positive ion mode

```{r}
cols<-rep(c("darkcyan","darkorange3"),each=3)
boxplot(pos.annotDf[,9:22],col=cols,main="after median normalization in positive ion mode")
colSums(pos.annotDf[,9:22])
```

```{r}
cor_plot <- pheatmap::pheatmap(cor(log2(pos.annotDf[,9:22])), main = "after median normalization in positive ion mode")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/after-norm-correlation-plot.png", plot = cor_plot,width = 10, height = 10, units = "in")
```

negative ion mode

```{r}
boxplot(neg.annotDf[,9:22],col=cols,main="after median normalization in negative ion mode")
colSums(neg.annotDf[,9:22])
```

```{r}
cor_plot <- pheatmap::pheatmap(cor(log2(neg.annotDf[,9:22])), main = "after median normalization in negative ion mode")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/after-norm-correlation-plot.png", plot = cor_plot,width = 10, height = 10, units = "in")

rm(cor_plot)
```

## 3.3 Annotation for lipid categories and InChIKey number

### positive ion mode

```{r}
# R-based tools for standardization of metabolite names (RefMet)
# https://github.com/metabolomicsworkbench/RefMet
# reference: https://rdcu.be/caRk5

# generate text file that the package requires
#write.table(pos.annotDf$Name, "output/vignette_run_outliersremoved_pos/lipids in positive ion mode.txt",quote = FALSE, row.names = FALSE, col.names = FALSE, sep = "\t") # mannually check whether lines are intact, add first line as a heading e.g. "NAME", "METABOLITE_NAME", "ID", etc


#** hide it only for output purpose

#jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_pos/lipids in positive ion mode.txt"
#RefMet_mapped <- refmet_map(jobDir)
#head(RefMet_mapped[,1:4])

#RefMet_mapped$inchi_key <- numeric(nrow(RefMet_mapped))
# Use a data frame column containing metabolite names as input
#for (i in 1:233){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 234 has an error

#for (i in 235:466){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 467 has an error

#metadata <- refmet_metadata(RefMet_mapped[468, "Standardized.name"])
#  RefMet_mapped$inchi_key[468] <- metadata[1,"inchi_key"] #line 469 has an error

#for (i in 470:673){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 674 has an error
  
#for (i in 675:nrow(RefMet_mapped)){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 674 has an error  

#write.csv(RefMet_mapped, "output/vignette_run_outliersremoved_pos/reference_map.csv") 

# manually annotate most lipids through lipid maps
pos.RefMet_mapped <- read.csv("output/vignette_run_outliersremoved_pos/manual-reference_map.csv")

# combine this with DEG analysis
pos.RefMet_mapped <- left_join(pos.RefMet_mapped,pos.annotDf, by = c("Input.name" = "Name"))
rownames(pos.RefMet_mapped[is.na(pos.RefMet_mapped$`Molino-Surface_fish_PValue`),])

#There are a few NAs need to be corrected
pos.RefMet_mapped[3,1] <- pos.annotDf[3,1]
pos.RefMet_mapped[3, 9:29] <- pos.annotDf[3, 2:22]

pos.RefMet_mapped[27,1] <- pos.annotDf[27,1]
pos.RefMet_mapped[27, 9:29] <- pos.annotDf[27, 2:22]

pos.RefMet_mapped[184,1] <- pos.annotDf[184,1]
pos.RefMet_mapped[184, 9:29] <- pos.annotDf[184, 2:22]

rownames(pos.RefMet_mapped[is.na(pos.RefMet_mapped$`Molino-Surface_fish_PValue`),])# no NAs

# delete mass and formula column
pos.RefMet_mapped <- pos.RefMet_mapped[, c(1:2, 5:29)]
#write.csv(pos.RefMet_mapped, "output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/all lipids in positive ion mode.csv")

rm(pos.jobDir)
```


### negative ion mode

```{r}
# R-based tools for standardization of metabolite names (RefMet)
# https://github.com/metabolomicsworkbench/RefMet
# reference: https://rdcu.be/caRk5

# generate text file that the package requires
#write.table(neg.annotDf$Name, "output/vignette_run_outliersremoved_neg/lipids in negative ion mode.txt",quote = FALSE, row.names = FALSE, col.names = FALSE, sep = "\t") # mannually check whether lines are intact, add first line as a heading e.g. "NAME", "METABOLITE_NAME", "ID", etc

#** hide it only for output purpose

#jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/lipids in negative ion mode.txt"
#RefMet_mapped <- refmet_map(jobDir)
#head(RefMet_mapped[,1:4])

#RefMet_mapped$inchi_key <- numeric(nrow(RefMet_mapped))
#Use a data frame column containing metabolite names as input
#for (i in 1:28){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 29 has an error

#for (i in 30:131){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 132 and 133 has an error

#for (i in 134:142){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 143 has an error

#for (i in 144:236){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 237 has an error

#for (i in 238:670){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 671 has an error

#for (i in 672:690){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 691 has an error

  
#for (i in 692:nrow(RefMet_mapped)){
#  metadata <- refmet_metadata(RefMet_mapped[i, "Standardized.name"])
#  RefMet_mapped$inchi_key[i] <- metadata[1,"inchi_key"]
#} #line 691 has an error  

#write.csv(RefMet_mapped, "output/vignette_run_outliersremoved_neg/reference_map.csv") # need a lot of manual annotation

# manually annotate most lipids through lipid maps
neg.RefMet_mapped <- read.csv("output/vignette_run_outliersremoved_neg/manual-reference_map.csv")

# combine this with DEG analysis
neg.RefMet_mapped <- left_join(neg.RefMet_mapped,neg.annotDf, by = c("Input.name" = "Name"))
rownames(pos.RefMet_mapped[is.na(pos.RefMet_mapped$`Molino-Surface_fish_PValue`),])

#There are a few NAs need to be corrected
neg.RefMet_mapped[2,1] <- neg.annotDf[2,1]
neg.RefMet_mapped[2, 9:29] <- neg.annotDf[2, 2:22]

neg.RefMet_mapped[3,1] <- neg.annotDf[3,1]
neg.RefMet_mapped[3, 9:29] <- neg.annotDf[3, 2:22]

neg.RefMet_mapped[6,1] <- neg.annotDf[6,1]
neg.RefMet_mapped[6, 9:29] <- neg.annotDf[6, 2:22]

neg.RefMet_mapped[718,1] <- neg.annotDf[718,1]
neg.RefMet_mapped[718, 9:29] <- neg.annotDf[718, 2:22]

neg.RefMet_mapped[730,1] <- neg.annotDf[730,1]
neg.RefMet_mapped[730, 9:29] <- neg.annotDf[730, 2:22]

rownames(neg.RefMet_mapped[is.na(neg.RefMet_mapped$`Molino-Surface_fish_PValue`),])# no NAs

# delete mass and formula column
neg.RefMet_mapped <- neg.RefMet_mapped[, c(1:2, 5:29)]
#write.csv(neg.RefMet_mapped, "output/vignette_run_outliersremoved_neg/median-norm/DEG_analysis/all lipids in negative ion mode.csv")

rm(neg.jobDir)

```



## 3.4 results from DEG analysis

### positive ion mode
```{r}
jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/"

pos.up_cav_mapped <- pos.RefMet_mapped[pos.RefMet_mapped$`Molino-Surface_fish_AdjPVal` < 0.05 & pos.RefMet_mapped$`Pachón-Surface_fish_AdjPVal` < 0.05 & pos.RefMet_mapped$`Molino-Surface_fish_log2FoldChange` > 0.5 & pos.RefMet_mapped$`Pachón-Surface_fish_log2FoldChange` >0.5, ]

pos.down_cav_mapped <- pos.RefMet_mapped[pos.RefMet_mapped$`Molino-Surface_fish_AdjPVal` < 0.05 & pos.RefMet_mapped$`Pachón-Surface_fish_AdjPVal` < 0.05 & pos.RefMet_mapped$`Molino-Surface_fish_log2FoldChange` < -0.5 & pos.RefMet_mapped$`Pachón-Surface_fish_log2FoldChange` < -0.5, ]

#write.csv(pos.up_cav_mapped, file=paste(jobDir, "mapped lipids upregulated in cavefish in positive ion mode.csv", sep="/"))
#write.csv(pos.down_cav_mapped, file=paste(jobDir, "mapped lipids downregulated in cavefish in positive ion mode.csv", sep="/"))

rm(jobDir)
```

### negative ion mode
```{r}
jobDir <- "/Volumes/projects/fx2482/lipidomics-2cellstage/output/vignette_run_outliersremoved_neg/median-norm/DEG_analysis/"

neg.up_cav_mapped <- neg.RefMet_mapped[neg.RefMet_mapped$`Molino-Surface_fish_AdjPVal` < 0.05 & neg.RefMet_mapped$`Pachón-Surface_fish_AdjPVal` < 0.05 & neg.RefMet_mapped$`Molino-Surface_fish_log2FoldChange` > 0.5 & neg.RefMet_mapped$`Pachón-Surface_fish_log2FoldChange` >0.5, ]

neg.down_cav_mapped <- neg.RefMet_mapped[neg.RefMet_mapped$`Molino-Surface_fish_AdjPVal` < 0.05 & neg.RefMet_mapped$`Pachón-Surface_fish_AdjPVal` < 0.05 & neg.RefMet_mapped$`Molino-Surface_fish_log2FoldChange` < -0.5 & neg.RefMet_mapped$`Pachón-Surface_fish_log2FoldChange` < -0.5, ]

#write.csv(neg.up_cav_mapped, file=paste(jobDir, "mapped lipids upregulated in cavefish in negitive ion mode.csv", sep="/"))
#write.csv(neg.down_cav_mapped, file=paste(jobDir, "mapped lipids downregulated in cavefish in negitive ion mode.csv", sep="/"))

rm(jobDir)
```



```{r}
rm(merged.neg,merged.neg.filtered,merged.pos,merged.pos.filtered,neg.annotDf,neg.down_cav_mapped,neg.filtered.sampleinfo,neg.RefMet_mapped,neg.up_cav_mapped,pos.annotDf,pos.down_cav_mapped,pos.filtered.sampleinfo,pos.RefMet_mapped,pos.up_cav_mapped)
```


# 4. Plots

## Figure 2D and SuppFig 2C PCA plot

```{r}
# create a sample info table
pos.annotDf <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/stat_table.csv")
neg.annotDf <- read.csv("output/vignette_run_outliersremoved_neg/median-norm/DEG_analysis/stat_table.csv")

pos.filtered.sampleinfo <- data.frame(
  sample = colnames(pos.annotDf[,10:23]),
  Population = c("Molino","Molino","Molino","Molino","Molino","Pachón","Pachón","Pachón","Pachón","Pachón","Surface fish","Surface fish","Surface fish","Surface fish"),
  stringsAsFactors = FALSE  # This ensures that text columns are not converted to factors
)

pos.filtered.sampleinfo$Population <- factor(pos.filtered.sampleinfo$Population, levels = c("Surface fish", "Pachón", "Molino"))

neg.filtered.sampleinfo <- data.frame(
  sample = colnames(neg.annotDf[,10:23]),
  Population = c("Molino","Molino","Molino","Molino","Molino","Pachón","Pachón","Pachón","Pachón","Pachón","Surface fish","Surface fish","Surface fish","Surface fish"),
  stringsAsFactors = FALSE  # This ensures that text columns are not converted to factors
)

neg.filtered.sampleinfo$Population <- factor(neg.filtered.sampleinfo$Population, levels = c("Surface fish", "Pachón", "Molino"))

```


```{r}
#from https://cran.r-project.org/web/packages/ggfortify/vignettes/plot_pca.html
#from https://bioinformatics-core-shared-training.github.io/Bulk_RNAseq_Course_2021/Markdowns/07_Data_Exploration.html#principal-component-analysis

#To run the PCA we should first normalize our data for library size and transform to a log scale. 

# positive ion mode
# read the norm data
normCounts <- pos.annotDf[,10:23]

# remove NA lines
filtered_normCounts <- normCounts[complete.cases(normCounts), ]

log_data <- log2(filtered_normCounts + 1)
pca_res <- prcomp(t(log_data), scale. = TRUE)
autoplot(pca_res, label = TRUE)

#read sampleinfo
pca_plot <- autoplot(pca_res, data = pos.filtered.sampleinfo,  colour='Population', size=1.8,
#                     frame = TRUE, frame.type = "norm"
                     )

pca_plot <- pca_plot + 
  scale_color_manual(values = pvg3)+
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = 0, linetype = "dashed") + # Add foldchange threshold
  theme_light()+
  labs(title = "Lipidomics positive ion mode - PCA plot") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5))  # Center the title

pca_plot

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2D Lipidomics positive ion mode - PCA plot (median-norm).png", plot = pca_plot, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2D Lipidomics positive ion mode - PCA plot (median-norm).svg", plot = pca_plot, width = 6, height = 4, units = "in")

rm(normCounts,filtered_normCounts,log_data,pca_res,pca_plot)









# negative ion mode

# read the norm data
normCounts <- neg.annotDf[,10:23]

# remove NA lines
filtered_normCounts <- normCounts[complete.cases(normCounts), ]

log_data <- log2(filtered_normCounts + 1)
pca_res <- prcomp(t(log_data), scale. = TRUE)
autoplot(pca_res, label = TRUE)

#read sampleinfo
pca_plot <- autoplot(pca_res, data = neg.filtered.sampleinfo,  colour='Population', size=1.8)

pca_plot <- pca_plot + 
  scale_color_manual(values = pvg3)+
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = 0, linetype = "dashed") + # Add foldchange threshold
  theme_light() +
  labs(title = "Lipidomics negative ion mode - PCA plot") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5))  # Center the title

pca_plot

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2C Lipidomics negative ion mode - PCA plot (median-norm).png", plot = pca_plot, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2C Lipidomics negative ion mode - PCA plot (median-norm).svg", plot = pca_plot, width = 6, height = 4, units = "in")

rm(normCounts,filtered_normCounts,log_data,pca_res,pca_plot)

```


## SuppFig 2D-G Volcano plot

```{r}

# positive ion mode
vol_plot_pa_sf <-ggplot(pos.annotDf, aes(x = `Pachón.Surface_fish_log2FoldChange`, y = -log10(`Pachón.Surface_fish_AdjPVal`))) +
  geom_point(color = ifelse(
    pos.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      pos.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 & 
      (pos.annotDf$`Pachón.Surface_fish_log2FoldChange` > 0.5 &
      pos.annotDf$`Molino.Surface_fish_log2FoldChange` > 0.5) |
      pos.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      pos.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 &
      (pos.annotDf$`Pachón.Surface_fish_log2FoldChange` < -0.5 &
      pos.annotDf$`Molino.Surface_fish_log2FoldChange` < -0.5), "red",
    "grey"), alpha = 0.6) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = c(-0.5, 0.5), linetype = "dashed") + # Add foldchange threshold
  theme_light()+
  labs(x = "Log2 Fold Change", y = "-log10(P adjusted)", title = "Pachón vs Surface fish") +
  theme(plot.title = element_text(hjust = 0.5)) 

vol_plot_pa_sf


vol_plot_mo_sf <- ggplot(pos.annotDf, aes(x = `Molino.Surface_fish_log2FoldChange`, y = -log10(`Molino.Surface_fish_AdjPVal`))) +
  geom_point(color = ifelse(
    pos.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      pos.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 & 
      (pos.annotDf$`Pachón.Surface_fish_log2FoldChange` > 0.5 &
      pos.annotDf$`Molino.Surface_fish_log2FoldChange` > 0.5) |
      pos.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      pos.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 &
      (pos.annotDf$`Pachón.Surface_fish_log2FoldChange` < -0.5 &
      pos.annotDf$`Molino.Surface_fish_log2FoldChange` < -0.5), "red",
    "grey"), alpha = 0.6) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = c(-0.5, 0.5), linetype = "dashed") + # Add foldchange threshold
  theme_light()+
  labs(x = "Log2 Fold Change", y = "-log10(P adjusted)", title = "Molino vs Surface fish") +
  theme(plot.title = element_text(hjust = 0.5)) 

vol_plot_mo_sf

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2D Pachón vs Surface fish - common lipids highlighted.png", plot = vol_plot_pa_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2D Pachón vs Surface fish - common lipids highlighted.svg", plot = vol_plot_pa_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2E Molino vs Surface fish - common genes highlighted.png", plot = vol_plot_mo_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2E Molino vs Surface fish - common genes highlighted.svg", plot = vol_plot_mo_sf, width = 6, height = 4, units = "in")

rm(vol_plot_mo_sf,vol_plot_pa_sf)











# negative ion mode
vol_plot_pa_sf <-ggplot(neg.annotDf, aes(x = `Pachón.Surface_fish_log2FoldChange`, y = -log10(`Pachón.Surface_fish_AdjPVal`))) +
  geom_point(color = ifelse(
    neg.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      neg.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 & 
      (neg.annotDf$`Pachón.Surface_fish_log2FoldChange` > 0.5 &
      neg.annotDf$`Molino.Surface_fish_log2FoldChange` > 0.5) |
      neg.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      neg.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 &
      (neg.annotDf$`Pachón.Surface_fish_log2FoldChange` < -0.5 &
      neg.annotDf$`Molino.Surface_fish_log2FoldChange` < -0.5), "red",
    "grey"), alpha = 0.6) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = c(-0.5, 0.5), linetype = "dashed") + # Add foldchange threshold
  theme_light()+
  labs(x = "Log2 Fold Change", y = "-log10(P adjusted)", title = "Pachón vs Surface fish") +
  theme(plot.title = element_text(hjust = 0.5)) 

vol_plot_pa_sf


vol_plot_mo_sf <- ggplot(neg.annotDf, aes(x = `Molino.Surface_fish_log2FoldChange`, y = -log10(`Molino.Surface_fish_AdjPVal`))) +
  geom_point(color = ifelse(
    neg.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      neg.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 & 
      (neg.annotDf$`Pachón.Surface_fish_log2FoldChange` > 0.5 &
      neg.annotDf$`Molino.Surface_fish_log2FoldChange` > 0.5) |
      neg.annotDf$`Pachón.Surface_fish_AdjPVal` < 0.05 &
      neg.annotDf$`Molino.Surface_fish_AdjPVal` < 0.05 &
      (neg.annotDf$`Pachón.Surface_fish_log2FoldChange` < -0.5 &
      neg.annotDf$`Molino.Surface_fish_log2FoldChange` < -0.5), "red",
    "grey"), alpha = 0.6) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +  # Add significance threshold
  geom_vline(xintercept = c(-0.5, 0.5), linetype = "dashed") + # Add foldchange threshold
  theme_light()+
  labs(x = "Log2 Fold Change", y = "-log10(P adjusted)", title = "Molino vs Surface fish") +
  theme(plot.title = element_text(hjust = 0.5)) 

vol_plot_mo_sf

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2F Pachón vs Surface fish - common lipids highlighted.png", plot = vol_plot_pa_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2F Pachón vs Surface fish - common lipids highlighted.svg", plot = vol_plot_pa_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2G Molino vs Surface fish - common genes highlighted.png", plot = vol_plot_mo_sf, width = 6, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/SuppFig 2G Molino vs Surface fish - common genes highlighted.svg", plot = vol_plot_mo_sf, width = 6, height = 4, units = "in")

rm(vol_plot_mo_sf,vol_plot_pa_sf)

```


## 4.3 Bar plot for specific lipid species

### SuppFig 2H Astaxanthin

```{r}

Astaxanthin <- pos.annotDf[pos.annotDf$Name == "Astaxanthin", 10:23]

Astaxanthin_df <- data.frame(
  Population = c("Molino","Molino","Molino","Molino","Molino",
            "Pachón","Pachón","Pachón","Pachón","Pachón",
            "Surface fish","Surface fish","Surface fish","Surface fish"),
  Replicate = c("Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
                "Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
                "Rep1", "Rep2", "Rep3", "Rep4"),
  Value = as.numeric(as.character(Astaxanthin))
)

Astaxanthin_df$Population <- factor(Astaxanthin_df$Population, levels = c("Surface fish", "Pachón", "Molino"))

# color from https://nanx.me/ggsci/reference/pal_npg.html

Astaxanthin_plot <- ggplot(Astaxanthin_df, aes(Population, Value, fill = Population)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg3) +
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistics
  geom_segment(
    aes(x = 1, xend = 2, y = max(Astaxanthin_df$Value) + 0.8, yend = max(Astaxanthin_df$Value) + 0.8),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = as.numeric(pos.annotDf[pos.annotDf$Name == "Astaxanthin", 4])) # PA vs SF
  ) +
  
  annotate(
    "text",
    x = 1.5, y = max(Astaxanthin_df$Value) + 0.9,
    label = paste("p =", format(as.numeric(pos.annotDf[pos.annotDf$Name == "Astaxanthin", 4]), digits = 2)),
    hjust = 0.5, vjust = -0.8,
    size = 4, color = "black") + # PA vs SF 
  
  theme_minimal() +
  labs(title = "Astaxanthin in 2-cell stage embryos", x = "Population", y = "Astaxanthin normalized abundance") +
  theme(text = element_text(size = 16)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) +  # Rotate x-axis labels by 30 degrees
  theme(plot.title = element_text(hjust = 0.5))  # Center the title


Astaxanthin_plot

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2H Astaxanthin in 2-cell stage embryos.png", plot = Astaxanthin_plot, width = 6, height = 6, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/SuppFig 2H Astaxanthin in 2-cell stage embryos.svg", plot = Astaxanthin_plot, width = 6, height = 6, units = "in")

rm(Astaxanthin, Astaxanthin_df, Astaxanthin_plot)

```


### 4.3.2 Anhydrovitamin A

```{r}

AnhydrovitaminA <- pos.annotDf[pos.annotDf$Name == "235BBF3K97", 9:22]

AnhydrovitaminA_df <- data.frame(
  Population = c("Molino","Molino","Molino","Molino","Molino",
            "Pachón","Pachón","Pachón","Pachón","Pachón",
            "Surface fish","Surface fish","Surface fish","Surface fish"),
  Replicate = c("Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
                "Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
                "Rep1", "Rep2", "Rep3", "Rep4"),
  Value = as.numeric(as.character(AnhydrovitaminA))
)

AnhydrovitaminA_df$Population <- factor(AnhydrovitaminA_df$Population, levels = c("Surface fish", "Pachón", "Molino"))

# color from https://nanx.me/ggsci/reference/pal_npg.html

AnhydrovitaminA_plot <- ggplot(AnhydrovitaminA_df, aes(Population, Value, fill = Population)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg3) +
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistics
  geom_segment(
    aes(x = 1, xend = 2, y = max(AnhydrovitaminA_df$Value) + 0.8, yend = max(AnhydrovitaminA_df$Value) + 0.8),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = as.numeric(pos.annotDf[pos.annotDf$Name == "235BBF3K97", 4])) # PA vs SF
  ) +
  geom_segment(
    aes(x = 1, xend = 3, y = max(AnhydrovitaminA_df$Value) + 1.5, yend = max(AnhydrovitaminA_df$Value) + 1.5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = as.numeric(pos.annotDf[pos.annotDf$Name == "235BBF3K97", 3])) # MO vs SF
  ) +
  
  annotate(
    "text",
    x = 1.5, y = max(AnhydrovitaminA_df$Value) + 0.9,
    label = paste("p =", format(as.numeric(pos.annotDf[pos.annotDf$Name == "235BBF3K97", 4]), digits = 2)),
    hjust = 0.5, vjust = -0.8,
    size = 4, color = "black") + # PA vs SF 
  annotate(
    "text",
    x = 2, y = max(AnhydrovitaminA_df$Value) + 1.6,
    label = paste("p =", format(as.numeric(pos.annotDf[pos.annotDf$Name == "235BBF3K97", 3]), digits = 2)),
    hjust = 0.5, vjust = -0.8,
    size = 4, color = "black") + # PA vs SF 
  
  theme_minimal() +
  ylab("Normalized abundance") +
  labs(title = "AnhydrovitaminA in 2-cell stage embryos", x = "Population", y = "Normalized abundance") +
  theme(text = element_text(size = 16)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) +  # Rotate x-axis labels by 30 degrees
  theme(plot.title = element_text(hjust = 0.5))  # Center the title


AnhydrovitaminA_plot

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Anhydrovitamin A in 2-cell stage embryos.png", plot = AnhydrovitaminA_plot, width = 6, height = 6, units = "in")

rm(AnhydrovitaminA, AnhydrovitaminA_df, AnhydrovitaminA_plot)

```

## Figure 2H GO term plot
```{r}
# got the GO term analysis from LION web
# http://www.lipidontology.com
# use upregulated lipids as target and all lipids as background

up.go.lipid <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/LION/LION-enrichment-job2.csv")

# select FDR q value < 0.05
up.go.lipid <- up.go.lipid[up.go.lipid$FDR.q.value<0.05, ]
up.go.lipid$`-LOG(FDR q-value)` <- -log10(up.go.lipid$FDR.q.value)
rownames(up.go.lipid) <- up.go.lipid$Discription

p <- ggplot(up.go.lipid, aes(x = `-LOG(FDR q-value)`, y = reorder(Discription, -p.value))) +
  geom_bar(stat = "identity", fill = "#E87A70",color = "black") + # help to do the horizontal plot
  labs(
    title = "GO-Term analysis for cavefish upregulated lipids",
    x = "-LOG(FDR q-value)",
    y = "GO Term"
  ) +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12, 
                                   #angle = 45, 
                                   hjust = 1),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))


p

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2H GO-Term analysis for cavefish upregulated lipids.png", plot = p, width = 7, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2H GO-Term analysis for cavefish upregulated lipids.svg", plot = p, width = 7, height = 4, units = "in")

rm(up.go.lipid,p)




down.go.lipid <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/LION/down_cav_vssf/LION-enrichment-job2.csv")

# select FDR q value < 0.05
down.go.lipid <- down.go.lipid[1:8, ]
down.go.lipid$`-LOG(FDR q-value)` <- -log10(down.go.lipid$FDR.q.value)
rownames(down.go.lipid) <- down.go.lipid$Discription

p <- ggplot(down.go.lipid, aes(x = `-LOG(FDR q-value)`, y = reorder(Discription, -p.value))) +
  geom_bar(stat = "identity", fill = "grey",color = "black") + # help to do the horizontal plot
  labs(
    title = "GO-Term analysis for cavefish downregulated lipids",
    x = "-LOG(FDR q-value)",
    y = "GO Term"
  ) +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12, 
                                   #angle = 45, 
                                   hjust = 1),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))


p

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2H GO-Term analysis for cavefish downregulated lipids.png", plot = p, width = 7, height = 4, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2H GO-Term analysis for cavefish downregulated lipids.svg", plot = p, width = 7, height = 4, units = "in")

rm(down.go.lipid,p)
```



## 4.5 Category - Super class

## Positive ion mode

```{r}
pos.RefMet_mapped <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/all lipids in positive ion mode.csv")
```

### 4.4.1 Molino positive ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_MO_pos_1 <- aggregate(Group.Area..M1_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_pos_1 <- grouped_MO_pos_1 %>%
mutate(mo1.percent = Group.Area..M1_pos / sum(Group.Area..M1_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart_1 <- 
  ggplot(grouped_MO_pos_1, aes(x = "", y = Group.Area..M1_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 1 in positive ion mode", hjust = 0.5)

MO_pos_pie_chart_1









# Group by category and calculate the sum of values in other columns
grouped_MO_pos_2 <- aggregate(Group.Area..M2_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_pos_2 <- grouped_MO_pos_2 %>%
mutate(mo2.percent = Group.Area..M2_pos / sum(Group.Area..M2_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart_2 <- 
  ggplot(grouped_MO_pos_2, aes(x = "", y = Group.Area..M2_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 2 in positive ion mode", hjust = 0.5)

MO_pos_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_MO_pos_3 <- aggregate(Group.Area..M3_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_pos_3 <- grouped_MO_pos_3 %>%
mutate(mo3.percent = Group.Area..M3_pos / sum(Group.Area..M3_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart_3 <- 
  ggplot(grouped_MO_pos_3, aes(x = "", y = Group.Area..M3_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 3 in positive ion mode", hjust = 0.5)

MO_pos_pie_chart_3









# Group by category and calculate the sum of values in other columns
grouped_MO_pos_4 <- aggregate(Group.Area..M4_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_pos_4 <- grouped_MO_pos_4 %>%
mutate(mo4.percent = Group.Area..M4_pos / sum(Group.Area..M4_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart_4 <- 
  ggplot(grouped_MO_pos_4, aes(x = "", y = Group.Area..M4_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 4 in positive ion mode", hjust = 0.5)

MO_pos_pie_chart_4









# Group by category and calculate the sum of values in other columns
grouped_MO_pos_6 <- aggregate(Group.Area..M6_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_pos_6 <- grouped_MO_pos_6 %>%
mutate(mo6.percent = Group.Area..M6_pos / sum(Group.Area..M6_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart_6 <- 
  ggplot(grouped_MO_pos_6, aes(x = "", y = Group.Area..M6_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 6 in positive ion mode", hjust = 0.5)

MO_pos_pie_chart_6


MO_combined_plot <- grid.arrange(MO_pos_pie_chart_1, MO_pos_pie_chart_2,
                                 MO_pos_pie_chart_3, MO_pos_pie_chart_4,
                                 MO_pos_pie_chart_6, nrow = 2)

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/MO_positive_pie_chart_lipidCategory.png", plot = MO_combined_plot, width = 20, height = 12, units = "in")

rm(MO_pos_pie_chart_1,MO_pos_pie_chart_2,MO_pos_pie_chart_3,MO_pos_pie_chart_4,MO_pos_pie_chart_6,MO_combined_plot)
```

### 4.4.2 Pachón positive ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_PA_pos_1 <- aggregate(Group.Area..P1_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_pos_1 <- grouped_PA_pos_1 %>%
mutate(pa1.percent = Group.Area..P1_pos / sum(Group.Area..P1_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart_1 <- 
  ggplot(grouped_PA_pos_1, aes(x = "", y = Group.Area..P1_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 1 in positive ion mode", hjust = 0.5)

PA_pos_pie_chart_1









# Group by category and calculate the sum of values in other columns
grouped_PA_pos_2 <- aggregate(Group.Area..P2_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_pos_2 <- grouped_PA_pos_2 %>%
mutate(pa2.percent = Group.Area..P2_pos / sum(Group.Area..P2_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart_2 <- 
  ggplot(grouped_PA_pos_2, aes(x = "", y = Group.Area..P2_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 2 in positive ion mode", hjust = 0.5)

PA_pos_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_PA_pos_3 <- aggregate(Group.Area..P3_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_pos_3 <- grouped_PA_pos_3 %>%
mutate(pa3.percent = Group.Area..P3_pos / sum(Group.Area..P3_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart_3 <- 
  ggplot(grouped_PA_pos_3, aes(x = "", y = Group.Area..P3_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 3 in positive ion mode", hjust = 0.5)

PA_pos_pie_chart_3









# Group by category and calculate the sum of values in other columns
grouped_PA_pos_5 <- aggregate(Group.Area..P5_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_pos_5 <- grouped_PA_pos_5 %>%
mutate(pa5.percent = Group.Area..P5_pos / sum(Group.Area..P5_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart_5 <- 
  ggplot(grouped_PA_pos_5, aes(x = "", y = Group.Area..P5_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 5 in positive ion mode", hjust = 0.5)

PA_pos_pie_chart_5









# Group by category and calculate the sum of values in other columns
grouped_PA_pos_6 <- aggregate(Group.Area..P6_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_pos_6 <- grouped_PA_pos_6 %>%
mutate(pa6.percent = Group.Area..P6_pos / sum(Group.Area..P6_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart_6 <- 
  ggplot(grouped_PA_pos_6, aes(x = "", y = Group.Area..P6_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 6 in positive ion mode", hjust = 0.5)

PA_pos_pie_chart_6


PA_combined_plot <- grid.arrange(PA_pos_pie_chart_1, PA_pos_pie_chart_2,
                                 PA_pos_pie_chart_3, PA_pos_pie_chart_5,
                                 PA_pos_pie_chart_6, nrow = 2)

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/PA_positive_pie_chart_lipidCategory.png", plot = PA_combined_plot, width = 20, height = 12, units = "in")

rm(PA_pos_pie_chart_1,PA_pos_pie_chart_2,PA_pos_pie_chart_3,PA_pos_pie_chart_5,PA_pos_pie_chart_6,PA_combined_plot)

```


### 4.4.3 Surface fish positive ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_SF_pos_2 <- aggregate(Group.Area..S2_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_pos_2 <- grouped_SF_pos_2 %>%
mutate(sf2.percent = Group.Area..S2_pos / sum(Group.Area..S2_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_pos_pie_chart_2 <- 
  ggplot(grouped_SF_pos_2, aes(x = "", y = Group.Area..S2_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 2 in positive ion mode", hjust = 0.5)

SF_pos_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_SF_pos_4 <- aggregate(Group.Area..S4_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_pos_4 <- grouped_SF_pos_4 %>%
mutate(sf4.percent = Group.Area..S4_pos / sum(Group.Area..S4_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_pos_pie_chart_4 <- 
  ggplot(grouped_SF_pos_4, aes(x = "", y = Group.Area..S4_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 4 in positive ion mode", hjust = 0.5)

SF_pos_pie_chart_4









# Group by category and calculate the sum of values in other columns
grouped_SF_pos_5 <- aggregate(Group.Area..S5_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_pos_5 <- grouped_SF_pos_5 %>%
mutate(sf5.percent = Group.Area..S5_pos / sum(Group.Area..S5_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_pos_pie_chart_5 <- 
  ggplot(grouped_SF_pos_5, aes(x = "", y = Group.Area..S5_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 5 in positive ion mode", hjust = 0.5)

SF_pos_pie_chart_5









# Group by category and calculate the sum of values in other columns
grouped_SF_pos_6 <- aggregate(Group.Area..S6_pos ~ Super.class, data = pos.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_pos_6 <- grouped_SF_pos_6 %>%
mutate(sf6.percent = Group.Area..S6_pos / sum(Group.Area..S6_pos) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_pos_pie_chart_6 <- 
  ggplot(grouped_SF_pos_6, aes(x = "", y = Group.Area..S6_pos, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 6 in positive ion mode", hjust = 0.5)

SF_pos_pie_chart_6


SF_combined_plot <- grid.arrange(SF_pos_pie_chart_2,SF_pos_pie_chart_4,
                                 SF_pos_pie_chart_5,SF_pos_pie_chart_6,
                                 nrow = 2)

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/SF_positive_pie_chart_lipidCategory.png", plot = SF_combined_plot, width = 20, height = 12, units = "in")

rm(SF_pos_pie_chart_2,SF_pos_pie_chart_4,SF_pos_pie_chart_5,SF_pos_pie_chart_6,SF_combined_plot)

```


### 4.4.4 Positive ion summary - pie and barplot

```{r}

# combine all datasets
pos.summary <-  grouped_SF_pos_2 %>%
  left_join(grouped_SF_pos_4, by = "Super.class") %>%
  left_join(grouped_SF_pos_5, by = "Super.class") %>%
  left_join(grouped_SF_pos_6, by = "Super.class") %>%
  left_join(grouped_PA_pos_1, by = "Super.class") %>%
  left_join(grouped_PA_pos_2, by = "Super.class") %>%
  left_join(grouped_PA_pos_3, by = "Super.class") %>%
  left_join(grouped_PA_pos_5, by = "Super.class") %>%
  left_join(grouped_PA_pos_6, by = "Super.class") %>%
  left_join(grouped_MO_pos_1, by = "Super.class") %>%
  left_join(grouped_MO_pos_2, by = "Super.class") %>%
  left_join(grouped_MO_pos_3, by = "Super.class") %>%
  left_join(grouped_MO_pos_4, by = "Super.class") %>%
  left_join(grouped_MO_pos_6, by = "Super.class")

rm(grouped_SF_pos_2,grouped_SF_pos_4,grouped_SF_pos_5,grouped_SF_pos_6,grouped_PA_pos_1,grouped_PA_pos_2,grouped_PA_pos_3,grouped_PA_pos_5,grouped_PA_pos_6,grouped_MO_pos_1,grouped_MO_pos_2,grouped_MO_pos_3,grouped_MO_pos_4,grouped_MO_pos_6)
```

#### Figure 2E Pie chart for summary data of positive ion mode

```{r}

# clean the data
pos.pie <- pos.summary
pos.pie <- pos.pie[,c(1,2,4,6,8,10,12,14,16,18,20,22,24,26,28)]

pos.pie$SF <- rowSums(pos.pie[, grep("^Group.Area..S\\d+", colnames(pos.pie), value = TRUE)], na.rm = TRUE)
pos.pie$PA <- rowSums(pos.pie[, grep("^Group.Area..P", colnames(pos.pie), value = TRUE)], na.rm = TRUE)
pos.pie$MO <- rowSums(pos.pie[, grep("^Group.Area..M", colnames(pos.pie), value = TRUE)], na.rm = TRUE)
pos.pie$all <- rowSums(pos.pie[, grep("^Group.Area..", colnames(pos.pie), value = TRUE)], na.rm = TRUE)

# remove individual values
pos.pie <- pos.pie[,c(1,16:19)]

# Surface fish
# Group by category and calculate the sum of values in other columns

pos.pie.sf <- aggregate(SF ~ Super.class, data = pos.pie, FUN = sum)

# Calculate percentages
pos.pie.sf <- pos.pie.sf %>%
mutate(percent = SF / sum(SF) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_pos_pie_chart <- 
  ggplot(pos.pie.sf, aes(x = "", y = SF, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish in positive ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

SF_pos_pie_chart







# Pachón
# Group by category and calculate the sum of values in other columns

pos.pie.pa <- aggregate(PA ~ Super.class, data = pos.pie, FUN = sum)

# Calculate percentages
pos.pie.pa <- pos.pie.pa %>%
mutate(percent = PA / sum(PA) * 100)

# Plot a pie chart
# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_pos_pie_chart <- 
  ggplot(pos.pie.pa, aes(x = "", y = PA, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón cavefish in positive ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

PA_pos_pie_chart








# Molino
# Group by category and calculate the sum of values in other columns

pos.pie.mo <- aggregate(MO ~ Super.class, data = pos.pie, FUN = sum)

# Calculate percentages
pos.pie.mo <- pos.pie.mo %>%
mutate(percent = MO / sum(MO) * 100)

# Plot a pie chart
# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_pos_pie_chart <- 
  ggplot(pos.pie.mo, aes(x = "", y = MO, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino cavefish in positive ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

MO_pos_pie_chart

combined_plot <- grid.arrange(SF_pos_pie_chart,PA_pos_pie_chart,
                                 MO_pos_pie_chart,nrow = 1)


#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/positive_all_lipidCategory.png", plot = combined_plot, width = 25, height = 12, units = "in")




# All
# Group by category and calculate the sum of values in other columns

pos.pie.all <- aggregate(all ~ Super.class, data = pos.pie, FUN = sum)

# Calculate percentages
pos.pie.all <- pos.pie.all %>%
mutate(percent = all / sum(all) * 100)

# Plot a pie chart
# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

All_pos_pie_chart <- 
  ggplot(pos.pie.all, aes(x = "", y = all, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  ggtitle(expression("Lipid catogories for " * italic("A. mexicanus") * " in positive ion mode")) +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

All_pos_pie_chart
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2E positive_all_lipidCategory.png", plot = All_pos_pie_chart, width = 8, height = 5, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2E positive_all_lipidCategory.svg", plot = All_pos_pie_chart, width = 8, height = 5, units = "in")

rm(pos.pie,pos.pie.sf,SF_pos_pie_chart,pos.pie.pa,PA_pos_pie_chart,pos.pie.mo,MO_pos_pie_chart,combined_plot,pos.pie.all,All_pos_pie_chart)
```

#### 4.4.4.2 Bar plot for summary data of positive ion mode

```{r}

pos.bar <- data.frame(
  Population = 0,
  Replicate = 0,
  Category = 0,
  Value = 0)

for (i in 1:18){
  pos.bar.temp <- data.frame(
  Population = c("Surface fish","Surface fish","Surface fish","Surface fish",
  "Pachón","Pachón","Pachón","Pachón","Pachón",
  "Molino","Molino","Molino","Molino","Molino"),
  Replicate = c("Rep1", "Rep2", "Rep3", "Rep4",
              "Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
              "Rep1", "Rep2", "Rep3", "Rep4", "Rep5"),
  Value = as.numeric(as.character(pos.summary[i,c(2,4,6,8,10,12,14,16,18,20,22,24,26,28)])))
  pos.bar.temp$Category <- pos.summary[i,"Super.class"]
  pos.bar <- rbind(pos.bar, pos.bar.temp)
}

# remove tempory 0 line
pos.bar <- pos.bar[2:253,]
pos.bar$Population <- factor(pos.bar$Population, levels = c("Surface fish", "Pachón", "Molino"))

pos.bar_plot <- ggplot(pos.bar, aes(x = Population, y = Value, fill = Category)) + 
  geom_bar(position = 'dodge', stat = 'summary', width = 0.9) +
  geom_errorbar(stat = 'summary', position = 'dodge', width = 0.9) +
  scale_fill_manual(values = pvg20) +  # Specify custom colors
  labs(title = "lipid abundance in positive ion mode in 2-cell stage embryos", x = "Morph", y = "Normalized abundance") +
  geom_point(aes(x = Population), shape = 21, position = position_dodge(width = 0.9)) +
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "lipid abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title

pos.bar_plot

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/positive_barplot_lipidCategory.png", plot = pos.bar_plot, width = 15, height = 6, units = "in")

rm(i,pos.bar.temp, pos.bar_plot)
```

#### 4.4.4.3 Comparison - super.class positive ion mode

##### Alkaloids

```{r}
pos.bar.Alkaloids <- pos.bar[pos.bar$Category == "Alkaloids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Alkaloids$Value, pos.bar.Alkaloids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Alkaloids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Alkaloids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Alkaloids <- ggplot(pos.bar.Alkaloids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[1]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Alkaloids$Value) + 5, yend = max(pos.bar.Alkaloids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Alkaloids$Value) + 10, yend = max(pos.bar.Alkaloids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Alkaloids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Alkaloids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Alkaloids normalized abundance") +
  labs(title = "Alkaloids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Alkaloids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Alkaloids_positive_boxplot.png", plot = Alkaloids, width = 6, height = 6, units = "in")

rm(pos.bar.Alkaloids,anova_test,tukey_result,Alkaloids)
```


##### Benzenoids

```{r}
pos.bar.Benzenoids <- pos.bar[pos.bar$Category == "Benzenoids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Benzenoids$Value, pos.bar.Benzenoids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Benzenoids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Benzenoids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Benzenoids <- ggplot(pos.bar.Benzenoids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[2]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Benzenoids$Value) + 5, yend = max(pos.bar.Benzenoids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Benzenoids$Value) + 10, yend = max(pos.bar.Benzenoids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Benzenoids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Benzenoids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Benzenoids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Benzenoids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Benzenoids_positive_boxplot.png", plot = Benzenoids, width = 6, height = 6, units = "in")

rm(pos.bar.Benzenoids,anova_test,tukey_result,Benzenoids)
```

##### Carbohydrates

```{r}
pos.bar.Carbohydrates <- pos.bar[pos.bar$Category == "Carbohydrates", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Carbohydrates$Value, pos.bar.Carbohydrates$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Carbohydrates) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Carbohydrates)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Carbohydrates <- ggplot(pos.bar.Carbohydrates, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[3]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Carbohydrates$Value) + 5, yend = max(pos.bar.Carbohydrates$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Carbohydrates$Value) + 10, yend = max(pos.bar.Carbohydrates$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Carbohydrates$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Carbohydrates$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Carbohydrates abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Carbohydrates

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Carbohydrates_positive_boxplot.png", plot = Carbohydrates, width = 6, height = 6, units = "in")

rm(pos.bar.Carbohydrates,anova_test,tukey_result,Carbohydrates)
```

##### Fatty Acyls

```{r}
pos.bar.FattyAcyls <- pos.bar[pos.bar$Category == "Fatty Acyls", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.FattyAcyls$Value, pos.bar.FattyAcyls$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.FattyAcyls) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.FattyAcyls)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

FattyAcyls <- ggplot(pos.bar.FattyAcyls, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[4]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.FattyAcyls$Value) + 50, yend = max(pos.bar.FattyAcyls$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.FattyAcyls$Value) + 100, yend = max(pos.bar.FattyAcyls$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.FattyAcyls$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.FattyAcyls$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Fatty Acyls abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
FattyAcyls

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Fatty Acyls_positive_boxplot.png", plot = FattyAcyls, width = 6, height = 6, units = "in")

rm(pos.bar.FattyAcyls,anova_test,tukey_result,FattyAcyls)
```

##### Glycerolipids

```{r}
pos.bar.Glycerolipids <- pos.bar[pos.bar$Category == "Glycerolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Glycerolipids$Value, pos.bar.Glycerolipids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Glycerolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Glycerolipids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Glycerolipids <- ggplot(pos.bar.Glycerolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[5]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Glycerolipids$Value) + 50, yend = max(pos.bar.Glycerolipids$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Glycerolipids$Value) + 100, yend = max(pos.bar.Glycerolipids$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Glycerolipids$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Glycerolipids$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Glycerolipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Glycerolipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Glycerolipids_positive_boxplot.png", plot = Glycerolipids, width = 6, height = 6, units = "in")

rm(pos.bar.Glycerolipids,anova_test,tukey_result,Glycerolipids)
```

##### Glycerophospholipids 

```{r}
pos.bar.Glycerophospholipids <- pos.bar[pos.bar$Category == "Glycerophospholipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Glycerophospholipids$Value, pos.bar.Glycerophospholipids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Glycerophospholipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Glycerophospholipids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Glycerophospholipids <- ggplot(pos.bar.Glycerophospholipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[6]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Glycerophospholipids$Value) + 50, yend = max(pos.bar.Glycerophospholipids$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Glycerophospholipids$Value) + 130, yend = max(pos.bar.Glycerophospholipids$Value) + 130),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Glycerophospholipids$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Glycerophospholipids$Value) + 150,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Glycerophospholipids normalized abundance") +
  labs(title = "Glycerophospholipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Glycerophospholipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Glycerophospholipids_positive_boxplot.png", plot = Glycerophospholipids, width = 6, height = 6, units = "in")

rm(pos.bar.Glycerophospholipids,anova_test,tukey_result,Glycerophospholipids)
```

##### Nucleic acids 

```{r}
pos.bar.Nucleicacids <- pos.bar[pos.bar$Category == "Nucleic acids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Nucleicacids$Value, pos.bar.Nucleicacids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Nucleicacids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Nucleicacids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Nucleicacids <- ggplot(pos.bar.Nucleicacids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[7]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Nucleicacids$Value) + 5, yend = max(pos.bar.Nucleicacids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Nucleicacids$Value) + 10, yend = max(pos.bar.Nucleicacids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Nucleicacids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Nucleicacids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Nucleic acids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Nucleicacids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Nucleic acids_positive_boxplot.png", plot = Nucleicacids, width = 6, height = 6, units = "in")

rm(pos.bar.Nucleicacids,anova_test,tukey_result,Nucleicacids)
```

##### Organic acids 

```{r}
pos.bar.Organicacids <- pos.bar[pos.bar$Category == "Organic acids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Organicacids$Value, pos.bar.Organicacids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Organicacids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Organicacids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organicacids <- ggplot(pos.bar.Organicacids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[8]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Organicacids$Value) + 10, yend = max(pos.bar.Organicacids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Organicacids$Value) + 20, yend = max(pos.bar.Organicacids$Value) + 20),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Organicacids$Value) + 13,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Organicacids$Value) + 23,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organic acids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organicacids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Organic acids_positive_boxplot.png", plot = Organicacids, width = 6, height = 6, units = "in")

rm(pos.bar.Organicacids,anova_test,tukey_result,Organicacids)
```

##### Organic nitrogen compounds 

```{r}
pos.bar.Organicnitrogencompounds <- pos.bar[pos.bar$Category == "Organic nitrogen compounds", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Organicnitrogencompounds$Value, pos.bar.Organicnitrogencompounds$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Organicnitrogencompounds) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Organicnitrogencompounds)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organicnitrogencompounds <- ggplot(pos.bar.Organicnitrogencompounds, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[9]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Organicnitrogencompounds$Value) + 1, yend = max(pos.bar.Organicnitrogencompounds$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Organicnitrogencompounds$Value) + 2, yend = max(pos.bar.Organicnitrogencompounds$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Organicnitrogencompounds$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Organicnitrogencompounds$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organic nitrogencompounds abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organicnitrogencompounds

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Organic nitrogencompounds_positive_boxplot.png", plot = Organicnitrogencompounds, width = 6, height = 6, units = "in")

rm(pos.bar.Organicnitrogencompounds,anova_test,tukey_result,Organicnitrogencompounds)
```

##### Organic oxygen compounds 

```{r}
pos.bar.Organicoxygencompounds <- pos.bar[pos.bar$Category == "Organic oxygen compounds", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Organicoxygencompounds$Value, pos.bar.Organicoxygencompounds$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Organicoxygencompounds) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Organicoxygencompounds)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organicoxygencompounds <- ggplot(pos.bar.Organicoxygencompounds, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[10]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Organicoxygencompounds$Value) + 1, yend = max(pos.bar.Organicoxygencompounds$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Organicoxygencompounds$Value) + 2, yend = max(pos.bar.Organicoxygencompounds$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Organicoxygencompounds$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Organicoxygencompounds$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organic oxygencompounds abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organicoxygencompounds

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Organic oxygencompounds_positive_boxplot.png", plot = Organicoxygencompounds, width = 6, height = 6, units = "in")

rm(pos.bar.Organicoxygencompounds,anova_test,tukey_result,Organicoxygencompounds)
```

##### Organoheterocyclic compounds 

```{r}
pos.bar.Organoheterocycliccompounds <- pos.bar[pos.bar$Category == "Organoheterocyclic compounds", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Organoheterocycliccompounds$Value, pos.bar.Organoheterocycliccompounds$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Organoheterocycliccompounds) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Organoheterocycliccompounds)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organoheterocycliccompounds <- ggplot(pos.bar.Organoheterocycliccompounds, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[11]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Organoheterocycliccompounds$Value) + 1, yend = max(pos.bar.Organoheterocycliccompounds$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Organoheterocycliccompounds$Value) + 2, yend = max(pos.bar.Organoheterocycliccompounds$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Organoheterocycliccompounds$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Organoheterocycliccompounds$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organoheterocyclic compounds abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organoheterocycliccompounds

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Organoheterocyclic compounds_positive_boxplot.png", plot = Organoheterocycliccompounds, width = 6, height = 6, units = "in")

rm(pos.bar.Organoheterocycliccompounds,anova_test,tukey_result,Organoheterocycliccompounds)
```

##### Other 

```{r}
pos.bar.Other <- pos.bar[pos.bar$Category == "Other", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Other$Value, pos.bar.Other$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Other) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Other)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Other <- ggplot(pos.bar.Other, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[13]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Other$Value) + 50, yend = max(pos.bar.Other$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Other$Value) + 100, yend = max(pos.bar.Other$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Other$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Other$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Other abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Other

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Other_positive_boxplot.png", plot = Other, width = 6, height = 6, units = "in")

rm(pos.bar.Other,anova_test,tukey_result,Other)
```

##### Polyketides 

```{r}
pos.bar.Polyketides <- pos.bar[pos.bar$Category == "Polyketides", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Polyketides$Value, pos.bar.Polyketides$Population, shapiro.test) # ~SF and PA not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Polyketides) # homogeneous

# Kruskal-Wallis Test
kruskal.test(Value ~ Population, data = pos.bar.Polyketides)

# Post-Hoc Tests
# Perform Dunn's Test for pairwise comparisons
dunn_test <- dunnTest(Value ~ Population, data = pos.bar.Polyketides, method = "bonferroni")
print(dunn_test)

Polyketides <- ggplot(pos.bar.Polyketides, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[14]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Polyketides$Value) + 5, yend = max(pos.bar.Polyketides$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = dunn_test$res$P.adj[3])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Polyketides$Value) + 10, yend = max(pos.bar.Polyketides$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = dunn_test$res$P.adj[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Polyketides$Value) + 7,
    label = paste("p = ", format.pval(dunn_test$res$P.adj[3], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Polyketides$Value) + 12,
    label = paste("p = ", format.pval(dunn_test$res$P.adj[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Polyketides abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Polyketides

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Polyketides_positive_boxplot.png", plot = Polyketides, width = 6, height = 6, units = "in")

rm(pos.bar.Polyketides,dunn_test,Polyketides)
```

##### Prenol Lipids 

```{r}
pos.bar.PrenolLipids <- pos.bar[pos.bar$Category == "Prenol Lipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.PrenolLipids$Value, pos.bar.PrenolLipids$Population, shapiro.test) # not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.PrenolLipids) # homogeneous

# Kruskal-Wallis Test
kruskal.test(Value ~ Population, data = pos.bar.PrenolLipids)

# Post-Hoc Tests
# Perform Dunn's Test for pairwise comparisons
dunn_test <- dunnTest(Value ~ Population, data = pos.bar.PrenolLipids, method = "bonferroni")
print(dunn_test)

PrenolLipids <- ggplot(pos.bar.PrenolLipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[15]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.PrenolLipids$Value) + 5, yend = max(pos.bar.PrenolLipids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = dunn_test$res$P.adj[3])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.PrenolLipids$Value) + 10, yend = max(pos.bar.PrenolLipids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = dunn_test$res$P.adj[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.PrenolLipids$Value) + 7,
    label = paste("p = ", format.pval(dunn_test$res$P.adj[3], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.PrenolLipids$Value) + 12,
    label = paste("p = ", format.pval(dunn_test$res$P.adj[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "PrenolLipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
PrenolLipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/PrenolLipids_positive_boxplot.png", plot = PrenolLipids, width = 6, height = 6, units = "in")

rm(pos.bar.PrenolLipids,dunn_test,PrenolLipids)
```

##### Saccharolipids

```{r}
pos.bar.Saccharolipids <- pos.bar[pos.bar$Category == "Saccharolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Saccharolipids$Value, pos.bar.Saccharolipids$Population, shapiro.test) # very close to 0.05, I will take it as normally distributed first

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Saccharolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Saccharolipids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Saccharolipids <- ggplot(pos.bar.Saccharolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[16]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Saccharolipids$Value) + 1, yend = max(pos.bar.Saccharolipids$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Saccharolipids$Value) + 2, yend = max(pos.bar.Saccharolipids$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Saccharolipids$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Saccharolipids$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Saccharolipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Saccharolipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Saccharolipids_positive_boxplot.png", plot = Saccharolipids, width = 6, height = 6, units = "in")

rm(pos.bar.Saccharolipids,anova_test,tukey_result,Saccharolipids)
```

##### Sphingolipids 

```{r}
pos.bar.Sphingolipids <- pos.bar[pos.bar$Category == "Sphingolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.Sphingolipids$Value, pos.bar.Sphingolipids$Population, shapiro.test) # Molino was a little on the edge..I will use one way anova

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.Sphingolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.Sphingolipids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Sphingolipids <- ggplot(pos.bar.Sphingolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[17]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.Sphingolipids$Value) + 50, yend = max(pos.bar.Sphingolipids$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.Sphingolipids$Value) + 100, yend = max(pos.bar.Sphingolipids$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.Sphingolipids$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.Sphingolipids$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Sphingolipids normalized abundance") +
  labs(title = "Sphingolipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Sphingolipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/Sphingolipids_positive_boxplot.png", plot = Sphingolipids, width = 6, height = 6, units = "in")

rm(pos.bar.Sphingolipids,anova_test,tukey_result,Sphingolipids)
```

##### Sterol Lipids 

```{r}
pos.bar.SterolLipids <- pos.bar[pos.bar$Category == "Sterol Lipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(pos.bar.SterolLipids$Value, pos.bar.SterolLipids$Population, shapiro.test) # not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = pos.bar.SterolLipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = pos.bar.SterolLipids)
# Perform Tukey's HSD test - posthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

SterolLipids <- ggplot(pos.bar.SterolLipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg20[18]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(pos.bar.SterolLipids$Value) + 5, yend = max(pos.bar.SterolLipids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(pos.bar.SterolLipids$Value) + 10, yend = max(pos.bar.SterolLipids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(pos.bar.SterolLipids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(pos.bar.SterolLipids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "SterolLipids abundance in positive ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
SterolLipids

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/all_categories/SterolLipids_positive_boxplot.png", plot = SterolLipids, width = 6, height = 6, units = "in")

rm(pos.bar.SterolLipids,anova_test,tukey_result,SterolLipids)
```


### 4.4.5 Positive ion DEG - pie and barplot

pie chart for up- and down-regulated lipid categories in cavefish from positive ion mode

```{r}
# For upregulated lipids in positive ion mode
pos.up_cav_mapped <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/mapped lipids upregulated in cavefish in positive ion mode.csv")

# calculate the percentage of each categories

name_percentage <- pos.up_cav_mapped %>%
  count(Super.class) %>%
  mutate(Percentage = n / sum(n) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

up_pos_pie_chart <- 
  ggplot(name_percentage, aes(x = "", y = n, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(Percentage), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = c("#A8BFBB","#D7CBB3","#8C7345","#D9AA1E","#BF4F26","#3F858C","#707322","#F2D43D")) +  # Specify custom colors
  labs(title = "Lipid categories for cavefish upregulated lipids in positive ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

up_pos_pie_chart

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2G Categories for cavefish upregulated lipids in positive ion mode.png", plot = up_pos_pie_chart, width = 8, height = 5, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2G Categories for cavefish upregulated lipids in positive ion mode.svg", plot = up_pos_pie_chart, width = 8, height = 5, units = "in")







# For downregulated lipids in positive ion mode
pos.down_cav_mapped <- read.csv("output/vignette_run_outliersremoved_pos/median-norm/DEG_analysis/mapped lipids downregulated in cavefish in positive ion mode.csv")

# calculate the percentage of each categories

name_percentage <- pos.down_cav_mapped %>%
  count(Super.class) %>%
  mutate(Percentage = n / sum(n) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

down_pos_pie_chart <- 
  ggplot(name_percentage, aes(x = "", y = n, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(Percentage), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = c("#1E4359","#58838C","#A8BFBB","#D7CBB3","#8C7345","#594A36","#D9BD9C","#8C4A32","#D9AA1E","#D98825","#BF4F26","#3F858C","#707322","#F2D43D")) +  # Specify custom colors
  labs(title = "Lipid categories for cavefish downregulated lipids in positive ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

down_pos_pie_chart

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2G Categories for cavefish downregulated lipids in positive ion mode.png", plot = down_pos_pie_chart, width = 8, height = 5, units = "in")
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2G Categories for cavefish downregulated lipids in positive ion mode.svg", plot = down_pos_pie_chart, width = 8, height = 5, units = "in")

#combined_plot <- grid.arrange(up_pos_pie_chart,down_pos_pie_chart,nrow = 2)

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/plots/Categories/Categories for cavefish up and downregulated lipids in positive ion mode.png", plot = combined_plot, width = 8, height = 10, units = "in")


rm(up_pos_pie_chart,down_pos_pie_chart,name_percentage,combined_plot,pos.up_cav_mapped,pos.down_cav_mapped)
```


## Negative ion mode

```{r}
neg.RefMet_mapped <- read.csv("output/vignette_run_outliersremoved_neg/median-norm/DEG_analysis/all lipids in negative ion mode.csv")
```

### 4.4.6 Molino negative ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_MO_neg_1 <- aggregate(Group.Area..M1_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_neg_1 <- grouped_MO_neg_1 %>%
mutate(mo1.percent = Group.Area..M1_neg / sum(Group.Area..M1_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart_1 <- 
  ggplot(grouped_MO_neg_1, aes(x = "", y = Group.Area..M1_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 1 in Negative ion mode", hjust = 0.5)

MO_neg_pie_chart_1









# Group by category and calculate the sum of values in other columns
grouped_MO_neg_2 <- aggregate(Group.Area..M2_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_neg_2 <- grouped_MO_neg_2 %>%
mutate(mo2.percent = Group.Area..M2_neg / sum(Group.Area..M2_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart_2 <- 
  ggplot(grouped_MO_neg_2, aes(x = "", y = Group.Area..M2_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 2 in Negative ion mode", hjust = 0.5)

MO_neg_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_MO_neg_3 <- aggregate(Group.Area..M3_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_neg_3 <- grouped_MO_neg_3 %>%
mutate(mo3.percent = Group.Area..M3_neg / sum(Group.Area..M3_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart_3 <- 
  ggplot(grouped_MO_neg_3, aes(x = "", y = Group.Area..M3_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 3 in Negative ion mode", hjust = 0.5)

MO_neg_pie_chart_3









# Group by category and calculate the sum of values in other columns
grouped_MO_neg_4 <- aggregate(Group.Area..M4_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_neg_4 <- grouped_MO_neg_4 %>%
mutate(mo4.percent = Group.Area..M4_neg / sum(Group.Area..M4_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart_4 <- 
  ggplot(grouped_MO_neg_4, aes(x = "", y = Group.Area..M4_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 4 in Negative ion mode", hjust = 0.5)

MO_neg_pie_chart_4









# Group by category and calculate the sum of values in other columns
grouped_MO_neg_6 <- aggregate(Group.Area..M6_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_MO_neg_6 <- grouped_MO_neg_6 %>%
mutate(mo6.percent = Group.Area..M6_neg / sum(Group.Area..M6_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart_6 <- 
  ggplot(grouped_MO_neg_6, aes(x = "", y = Group.Area..M6_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino replicate 6 in Negative ion mode", hjust = 0.5)

MO_neg_pie_chart_6


MO_combined_plot <- grid.arrange(MO_neg_pie_chart_1, MO_neg_pie_chart_2,
                                 MO_neg_pie_chart_3, MO_neg_pie_chart_4,
                                 MO_neg_pie_chart_6, nrow = 2)

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/MO_Negative_pie_chart_lipidCategory.png", plot = MO_combined_plot, width = 20, height = 12, units = "in")

rm(MO_neg_pie_chart_1,MO_neg_pie_chart_2,MO_neg_pie_chart_3,MO_neg_pie_chart_4,MO_neg_pie_chart_6,MO_combined_plot)
```

### 4.4.7 Pachón negative ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_PA_neg_1 <- aggregate(Group.Area..P1_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_neg_1 <- grouped_PA_neg_1 %>%
mutate(pa1.percent = Group.Area..P1_neg / sum(Group.Area..P1_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart_1 <- 
  ggplot(grouped_PA_neg_1, aes(x = "", y = Group.Area..P1_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 1 in Negative ion mode", hjust = 0.5)

PA_neg_pie_chart_1









# Group by category and calculate the sum of values in other columns
grouped_PA_neg_2 <- aggregate(Group.Area..P2_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_neg_2 <- grouped_PA_neg_2 %>%
mutate(pa2.percent = Group.Area..P2_neg / sum(Group.Area..P2_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart_2 <- 
  ggplot(grouped_PA_neg_2, aes(x = "", y = Group.Area..P2_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 2 in Negative ion mode", hjust = 0.5)

PA_neg_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_PA_neg_3 <- aggregate(Group.Area..P3_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_neg_3 <- grouped_PA_neg_3 %>%
mutate(pa3.percent = Group.Area..P3_neg / sum(Group.Area..P3_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart_3 <- 
  ggplot(grouped_PA_neg_3, aes(x = "", y = Group.Area..P3_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 3 in Negative ion mode", hjust = 0.5)

PA_neg_pie_chart_3









# Group by category and calculate the sum of values in other columns
grouped_PA_neg_5 <- aggregate(Group.Area..P5_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_neg_5 <- grouped_PA_neg_5 %>%
mutate(pa5.percent = Group.Area..P5_neg / sum(Group.Area..P5_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart_5 <- 
  ggplot(grouped_PA_neg_5, aes(x = "", y = Group.Area..P5_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 5 in Negative ion mode", hjust = 0.5)

PA_neg_pie_chart_5









# Group by category and calculate the sum of values in other columns
grouped_PA_neg_6 <- aggregate(Group.Area..P6_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_PA_neg_6 <- grouped_PA_neg_6 %>%
mutate(pa6.percent = Group.Area..P6_neg / sum(Group.Area..P6_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart_6 <- 
  ggplot(grouped_PA_neg_6, aes(x = "", y = Group.Area..P6_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón replicate 6 in Negative ion mode", hjust = 0.5)

PA_neg_pie_chart_6


PA_combined_plot <- grid.arrange(PA_neg_pie_chart_1, PA_neg_pie_chart_2,
                                 PA_neg_pie_chart_3, PA_neg_pie_chart_5,
                                 PA_neg_pie_chart_6, nrow = 2)

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/PA_Negative_pie_chart_lipidCategory.png", plot = PA_combined_plot, width = 20, height = 12, units = "in")

rm(PA_neg_pie_chart_1,PA_neg_pie_chart_2,PA_neg_pie_chart_3,PA_neg_pie_chart_5,PA_neg_pie_chart_6,PA_combined_plot)

```


### 4.4.8 Surface fish Negative ion mode

```{r}
# make plots and draw things

# Group by category and calculate the sum of values in other columns
grouped_SF_neg_2 <- aggregate(Group.Area..S2_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_neg_2 <- grouped_SF_neg_2 %>%
mutate(sf2.percent = Group.Area..S2_neg / sum(Group.Area..S2_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_neg_pie_chart_2 <- 
  ggplot(grouped_SF_neg_2, aes(x = "", y = Group.Area..S2_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 2 in Negative ion mode", hjust = 0.5)

SF_neg_pie_chart_2









# Group by category and calculate the sum of values in other columns
grouped_SF_neg_4 <- aggregate(Group.Area..S4_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_neg_4 <- grouped_SF_neg_4 %>%
mutate(sf4.percent = Group.Area..S4_neg / sum(Group.Area..S4_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_neg_pie_chart_4 <- 
  ggplot(grouped_SF_neg_4, aes(x = "", y = Group.Area..S4_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 4 in Negative ion mode", hjust = 0.5)

SF_neg_pie_chart_4









# Group by category and calculate the sum of values in other columns
grouped_SF_neg_5 <- aggregate(Group.Area..S5_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_neg_5 <- grouped_SF_neg_5 %>%
mutate(sf5.percent = Group.Area..S5_neg / sum(Group.Area..S5_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_neg_pie_chart_5 <- 
  ggplot(grouped_SF_neg_5, aes(x = "", y = Group.Area..S5_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 5 in Negative ion mode", hjust = 0.5)

SF_neg_pie_chart_5









# Group by category and calculate the sum of values in other columns
grouped_SF_neg_6 <- aggregate(Group.Area..S6_neg ~ Super.class, data = neg.RefMet_mapped, FUN = sum)

# Calculate percentages
grouped_SF_neg_6 <- grouped_SF_neg_6 %>%
mutate(sf6.percent = Group.Area..S6_neg / sum(Group.Area..S6_neg) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_neg_pie_chart_6 <- 
  ggplot(grouped_SF_neg_6, aes(x = "", y = Group.Area..S6_neg, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish replicate 6 in Negative ion mode", hjust = 0.5)

SF_neg_pie_chart_6


SF_combined_plot <- grid.arrange(SF_neg_pie_chart_2,SF_neg_pie_chart_4,
                                 SF_neg_pie_chart_5,SF_neg_pie_chart_6,
                                 nrow = 2)

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/SF_Negative_pie_chart_lipidCategory.png", plot = SF_combined_plot, width = 20, height = 12, units = "in")

rm(SF_neg_pie_chart_2,SF_neg_pie_chart_4,SF_neg_pie_chart_5,SF_neg_pie_chart_6,SF_combined_plot)

```


### 4.4.9 Negative ion summary - pie and barplot

```{r}

# combine all datasets
neg.summary <-  grouped_SF_neg_2 %>%
  left_join(grouped_SF_neg_4, by = "Super.class") %>%
  left_join(grouped_SF_neg_5, by = "Super.class") %>%
  left_join(grouped_SF_neg_6, by = "Super.class") %>%
  left_join(grouped_PA_neg_1, by = "Super.class") %>%
  left_join(grouped_PA_neg_2, by = "Super.class") %>%
  left_join(grouped_PA_neg_3, by = "Super.class") %>%
  left_join(grouped_PA_neg_5, by = "Super.class") %>%
  left_join(grouped_PA_neg_6, by = "Super.class") %>%
  left_join(grouped_MO_neg_1, by = "Super.class") %>%
  left_join(grouped_MO_neg_2, by = "Super.class") %>%
  left_join(grouped_MO_neg_3, by = "Super.class") %>%
  left_join(grouped_MO_neg_4, by = "Super.class") %>%
  left_join(grouped_MO_neg_6, by = "Super.class")

rm(grouped_SF_neg_2,grouped_SF_neg_4,grouped_SF_neg_5,grouped_SF_neg_6,grouped_PA_neg_1,grouped_PA_neg_2,grouped_PA_neg_3,grouped_PA_neg_5,grouped_PA_neg_6,grouped_MO_neg_1,grouped_MO_neg_2,grouped_MO_neg_3,grouped_MO_neg_4,grouped_MO_neg_6)
```

#### Figure 2E Pie chart for summary data of Negative ion mode

```{r}

# clean the data
neg.pie <- neg.summary
neg.pie <- neg.pie[,c(1,2,4,6,8,10,12,14,16,18,20,22,24,26,28)]

neg.pie$SF <- rowSums(neg.pie[, grep("^Group.Area..S\\d+", colnames(neg.pie), value = TRUE)], na.rm = TRUE)
neg.pie$PA <- rowSums(neg.pie[, grep("^Group.Area..P", colnames(neg.pie), value = TRUE)], na.rm = TRUE)
neg.pie$MO <- rowSums(neg.pie[, grep("^Group.Area..M", colnames(neg.pie), value = TRUE)], na.rm = TRUE)
neg.pie$all <- rowSums(neg.pie[, grep("^Group.Area..", colnames(neg.pie), value = TRUE)], na.rm = TRUE)

# remove individual values
neg.pie <- neg.pie[,c(1,16:19)]

# Surface fish
# Group by category and calculate the sum of values in other columns

neg.pie.sf <- aggregate(SF ~ Super.class, data = neg.pie, FUN = sum)

# Calculate percentages
neg.pie.sf <- neg.pie.sf %>%
mutate(percent = SF / sum(SF) * 100)

# Plot a pie chart

# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

SF_neg_pie_chart <- 
  ggplot(neg.pie.sf, aes(x = "", y = SF, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Surface fish in Negative ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

SF_neg_pie_chart







# Pachón
# Group by category and calculate the sum of values in other columns

neg.pie.pa <- aggregate(PA ~ Super.class, data = neg.pie, FUN = sum)

# Calculate percentages
neg.pie.pa <- neg.pie.pa %>%
mutate(percent = PA / sum(PA) * 100)

# Plot a pie chart
# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

PA_neg_pie_chart <- 
  ggplot(neg.pie.pa, aes(x = "", y = PA, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Pachón cavefish in Negative ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

PA_neg_pie_chart








# Molino
# Group by category and calculate the sum of values in other columns

neg.pie.mo <- aggregate(MO ~ Super.class, data = neg.pie, FUN = sum)

# Calculate percentages
neg.pie.mo <- neg.pie.mo %>%
mutate(percent = MO / sum(MO) * 100)

# Plot a pie chart
# position info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

MO_neg_pie_chart <- 
  ggplot(neg.pie.mo, aes(x = "", y = MO, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), position = position_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "Lipid catogories for Molino cavefish in Negative ion mode") +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

MO_neg_pie_chart

combined_plot <- grid.arrange(SF_neg_pie_chart,PA_neg_pie_chart,
                                 MO_neg_pie_chart,nrow = 1)


#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/Negative_all_lipidCategory.png", plot = combined_plot, width = 25, height = 12, units = "in")




# All
# Group by category and calculate the sum of values in other columns

neg.pie.all <- aggregate(all ~ Super.class, data = neg.pie, FUN = sum)

# Calculate percentages
neg.pie.all <- neg.pie.all %>%
mutate(percent = all / sum(all) * 100)

# Plot a pie chart
# negition info code from https://r-charts.com/part-whole/pie-chart-labels-outside-ggplot2/

All_neg_pie_chart <- 
  ggplot(neg.pie.all, aes(x = "", y = all, fill = Super.class)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar("y", start = 0) +
  theme_void() +
  #geom_text(aes(label = paste0(round(percent), "%")), negition = negition_stack(vjust = 0.5)) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  ggtitle(expression("Lipid catogories for " * italic("A. mexicanus") * " in negative ion mode")) +
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15))

All_neg_pie_chart
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/Figure 2E negative_all_lipidCategory.png", plot = All_neg_pie_chart, width = 8, height = 5, units = "in")
#ggsave("output/vignette_run_outliersremoved_neg/median-norm/paper/Figure 2E negative_all_lipidCategory.svg", plot = All_neg_pie_chart, width = 8, height = 5, units = "in")


rm(neg.pie,neg.pie.sf,SF_neg_pie_chart,neg.pie.pa,PA_neg_pie_chart,neg.pie.mo,MO_neg_pie_chart,combined_plot,neg.pie.all,All_neg_pie_chart)
```



#### 4.4.9.2 Bar plot for summary data of Negative ion mode

```{r}

neg.bar <- data.frame(
  Population = 0,
  Replicate = 0,
  Category = 0,
  Value = 0)

for (i in 1:15){
  neg.bar.temp <- data.frame(
  Population = c("Surface fish","Surface fish","Surface fish","Surface fish",
  "Pachón","Pachón","Pachón","Pachón","Pachón",
  "Molino","Molino","Molino","Molino","Molino"),
  Replicate = c("Rep1", "Rep2", "Rep3", "Rep4",
              "Rep1", "Rep2", "Rep3", "Rep4", "Rep5",
              "Rep1", "Rep2", "Rep3", "Rep4", "Rep5"),
  Value = as.numeric(as.character(neg.summary[i,c(2,4,6,8,10,12,14,16,18,20,22,24,26,28)])))
  neg.bar.temp$Category <- neg.summary[i,"Super.class"]
  neg.bar <- rbind(neg.bar, neg.bar.temp)
}

# remove tempory 0 line
neg.bar <- neg.bar[2:211,]
neg.bar$Population <- factor(neg.bar$Population, levels = c("Surface fish", "Pachón", "Molino"))

neg.bar_plot <- ggplot(neg.bar, aes(x = Population, y = Value, fill = Category)) + 
  geom_bar(position = 'dodge', stat = 'summary', width = 0.9) +
  geom_errorbar(stat = 'summary', position = 'dodge', width = 0.9) +
  scale_fill_manual(values = pvg15) +  # Specify custom colors
  labs(title = "lipid abundance in Negative ion mode in 2-cell stage embryos", x = "Morph", y = "Normalized abundance") +
  geom_point(aes(x = Population), shape = 21, position = position_dodge(width = 0.9)) +
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "lipid abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title

neg.bar_plot

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/Negative_barplot_lipidCategory.png", plot = neg.bar_plot, width = 15, height = 6, units = "in")

rm(i,neg.bar.temp, neg.bar_plot)
```

#### 4.4.9.3 Comparison - super.class Negative ion mode

##### Alkaloids

```{r}
neg.bar.Alkaloids <- neg.bar[neg.bar$Category == "Alkaloids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Alkaloids$Value, neg.bar.Alkaloids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Alkaloids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Alkaloids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Alkaloids <- ggplot(neg.bar.Alkaloids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[1]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Alkaloids$Value) + 1, yend = max(neg.bar.Alkaloids$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Alkaloids$Value) + 2, yend = max(neg.bar.Alkaloids$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Alkaloids$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Alkaloids$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Alkaloids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Alkaloids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Alkaloids_Negative_boxplot.png", plot = Alkaloids, width = 6, height = 6, units = "in")

rm(neg.bar.Alkaloids,anova_test,tukey_result,Alkaloids)
```


##### Benzenoids

```{r}
neg.bar.Benzenoids <- neg.bar[neg.bar$Category == "Benzenoids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Benzenoids$Value, neg.bar.Benzenoids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Benzenoids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Benzenoids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Benzenoids <- ggplot(neg.bar.Benzenoids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[2]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Benzenoids$Value) + 1, yend = max(neg.bar.Benzenoids$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Benzenoids$Value) + 2, yend = max(neg.bar.Benzenoids$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Benzenoids$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Benzenoids$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Benzenoids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Benzenoids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Benzenoids_Negative_boxplot.png", plot = Benzenoids, width = 6, height = 6, units = "in")

rm(neg.bar.Benzenoids,anova_test,tukey_result,Benzenoids)
```

##### Carbohydrates

```{r}
neg.bar.Carbohydrates <- neg.bar[neg.bar$Category == "Carbohydrates", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Carbohydrates$Value, neg.bar.Carbohydrates$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Carbohydrates) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Carbohydrates)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Carbohydrates <- ggplot(neg.bar.Carbohydrates, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[3]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Carbohydrates$Value) + 5, yend = max(neg.bar.Carbohydrates$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Carbohydrates$Value) + 10, yend = max(neg.bar.Carbohydrates$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Carbohydrates$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Carbohydrates$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Carbohydrates abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Carbohydrates

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Carbohydrates_Negative_boxplot.png", plot = Carbohydrates, width = 6, height = 6, units = "in")

rm(neg.bar.Carbohydrates,anova_test,tukey_result,Carbohydrates)
```

##### Fatty Acyls

```{r}
neg.bar.FattyAcyls <- neg.bar[neg.bar$Category == "Fatty Acyls", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.FattyAcyls$Value, neg.bar.FattyAcyls$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.FattyAcyls) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.FattyAcyls)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

FattyAcyls <- ggplot(neg.bar.FattyAcyls, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[4]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.FattyAcyls$Value) + 50, yend = max(neg.bar.FattyAcyls$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.FattyAcyls$Value) + 100, yend = max(neg.bar.FattyAcyls$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.FattyAcyls$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.FattyAcyls$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Fatty Acyls abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
FattyAcyls

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Fatty Acyls_Negative_boxplot.png", plot = FattyAcyls, width = 6, height = 6, units = "in")

rm(neg.bar.FattyAcyls,anova_test,tukey_result,FattyAcyls)
```

##### Glycerolipids

```{r}
neg.bar.Glycerolipids <- neg.bar[neg.bar$Category == "Glycerolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Glycerolipids$Value, neg.bar.Glycerolipids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Glycerolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Glycerolipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Glycerolipids <- ggplot(neg.bar.Glycerolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[5]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Glycerolipids$Value) + 5, yend = max(neg.bar.Glycerolipids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Glycerolipids$Value) + 10, yend = max(neg.bar.Glycerolipids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Glycerolipids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Glycerolipids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Glycerolipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Glycerolipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Glycerolipids_Negative_boxplot.png", plot = Glycerolipids, width = 6, height = 6, units = "in")

rm(neg.bar.Glycerolipids,anova_test,tukey_result,Glycerolipids)
```

##### Glycerophospholipids 

```{r}
neg.bar.Glycerophospholipids <- neg.bar[neg.bar$Category == "Glycerophospholipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Glycerophospholipids$Value, neg.bar.Glycerophospholipids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Glycerophospholipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Glycerophospholipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Glycerophospholipids <- ggplot(neg.bar.Glycerophospholipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[6]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Glycerophospholipids$Value) + 50, yend = max(neg.bar.Glycerophospholipids$Value) + 50),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Glycerophospholipids$Value) + 100, yend = max(neg.bar.Glycerophospholipids$Value) + 100),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Glycerophospholipids$Value) + 70,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Glycerophospholipids$Value) + 120,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Glycerophospholipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Glycerophospholipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Glycerophospholipids_Negative_boxplot.png", plot = Glycerophospholipids, width = 6, height = 6, units = "in")

rm(neg.bar.Glycerophospholipids,anova_test,tukey_result,Glycerophospholipids)
```

##### Nucleic acids 

```{r}
neg.bar.Nucleicacids <- neg.bar[neg.bar$Category == "Nucleic acids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Nucleicacids$Value, neg.bar.Nucleicacids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Nucleicacids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Nucleicacids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Nucleicacids <- ggplot(neg.bar.Nucleicacids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[7]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Nucleicacids$Value) + 5, yend = max(neg.bar.Nucleicacids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Nucleicacids$Value) + 10, yend = max(neg.bar.Nucleicacids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Nucleicacids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Nucleicacids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Nucleic acids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Nucleicacids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Nucleic acids_Negative_boxplot.png", plot = Nucleicacids, width = 6, height = 6, units = "in")

rm(neg.bar.Nucleicacids,anova_test,tukey_result,Nucleicacids)
```

##### Organic acids 

```{r}
neg.bar.Organicacids <- neg.bar[neg.bar$Category == "Organic acids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Organicacids$Value, neg.bar.Organicacids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Organicacids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Organicacids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organicacids <- ggplot(neg.bar.Organicacids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[8]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Organicacids$Value) + 10, yend = max(neg.bar.Organicacids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Organicacids$Value) + 20, yend = max(neg.bar.Organicacids$Value) + 20),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Organicacids$Value) + 13,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Organicacids$Value) + 23,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organic acids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organicacids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Organic acids_Negative_boxplot.png", plot = Organicacids, width = 6, height = 6, units = "in")

rm(neg.bar.Organicacids,anova_test,tukey_result,Organicacids)
```

##### Organic oxygen compounds 

```{r}
neg.bar.Organicoxygencompounds <- neg.bar[neg.bar$Category == "Organic oxygen compounds", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Organicoxygencompounds$Value, neg.bar.Organicoxygencompounds$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Organicoxygencompounds) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Organicoxygencompounds)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Organicoxygencompounds <- ggplot(neg.bar.Organicoxygencompounds, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[9]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Organicoxygencompounds$Value) + 1, yend = max(neg.bar.Organicoxygencompounds$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Organicoxygencompounds$Value) + 2, yend = max(neg.bar.Organicoxygencompounds$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Organicoxygencompounds$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Organicoxygencompounds$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Organic oxygencompounds abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Organicoxygencompounds

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Organic oxygencompounds_Negative_boxplot.png", plot = Organicoxygencompounds, width = 6, height = 6, units = "in")

rm(neg.bar.Organicoxygencompounds,anova_test,tukey_result,Organicoxygencompounds)
```

##### Other 

```{r}
neg.bar.Other <- neg.bar[neg.bar$Category == "Other", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Other$Value, neg.bar.Other$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Other) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Other)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Other <- ggplot(neg.bar.Other, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[10]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Other$Value) + 10, yend = max(neg.bar.Other$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Other$Value) + 20, yend = max(neg.bar.Other$Value) + 20),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Other$Value) + 13,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Other$Value) + 23,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Other abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Other

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Other_Negative_boxplot.png", plot = Other, width = 6, height = 6, units = "in")

rm(neg.bar.Other,anova_test,tukey_result,Other)
```

##### Polyketides 

```{r}
neg.bar.Polyketides <- neg.bar[neg.bar$Category == "Polyketides", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Polyketides$Value, neg.bar.Polyketides$Population, shapiro.test) # ~SF and PA not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Polyketides) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Polyketides)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Polyketides <- ggplot(neg.bar.Polyketides, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[11]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Polyketides$Value) + 5, yend = max(neg.bar.Polyketides$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Polyketides$Value) + 10, yend = max(neg.bar.Polyketides$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Polyketides$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Polyketides$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Polyketides abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Polyketides

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Polyketides_Negative_boxplot.png", plot = Polyketides, width = 6, height = 6, units = "in")

rm(neg.bar.Polyketides,anova_test,tukey_result,Polyketides)
```

##### Prenol Lipids 

```{r}
neg.bar.PrenolLipids <- neg.bar[neg.bar$Category == "Prenol Lipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.PrenolLipids$Value, neg.bar.PrenolLipids$Population, shapiro.test) # not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.PrenolLipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.PrenolLipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

PrenolLipids <- ggplot(neg.bar.PrenolLipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[12]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.PrenolLipids$Value) + 5, yend = max(neg.bar.PrenolLipids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.PrenolLipids$Value) + 10, yend = max(neg.bar.PrenolLipids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.PrenolLipids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.PrenolLipids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "PrenolLipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
PrenolLipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/PrenolLipids_Negative_boxplot.png", plot = PrenolLipids, width = 6, height = 6, units = "in")

rm(neg.bar.PrenolLipids,anova_test,tukey_result,PrenolLipids)
```

##### Saccharolipids

```{r}
neg.bar.Saccharolipids <- neg.bar[neg.bar$Category == "Saccharolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Saccharolipids$Value, neg.bar.Saccharolipids$Population, shapiro.test) # normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Saccharolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Saccharolipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Saccharolipids <- ggplot(neg.bar.Saccharolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[13]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Saccharolipids$Value) + 1, yend = max(neg.bar.Saccharolipids$Value) + 1),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Saccharolipids$Value) + 2, yend = max(neg.bar.Saccharolipids$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Saccharolipids$Value) + 1.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Saccharolipids$Value) + 2.2,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Saccharolipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Saccharolipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Saccharolipids_Negative_boxplot.png", plot = Saccharolipids, width = 6, height = 6, units = "in")

rm(neg.bar.Saccharolipids,anova_test,tukey_result,Saccharolipids)
```

##### Sphingolipids 

```{r}
neg.bar.Sphingolipids <- neg.bar[neg.bar$Category == "Sphingolipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.Sphingolipids$Value, neg.bar.Sphingolipids$Population, shapiro.test) # Molino was a little on the edge..I will use one way anova

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.Sphingolipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.Sphingolipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

Sphingolipids <- ggplot(neg.bar.Sphingolipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[14]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.Sphingolipids$Value) + 5, yend = max(neg.bar.Sphingolipids$Value) + 5),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.Sphingolipids$Value) + 10, yend = max(neg.bar.Sphingolipids$Value) + 10),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.Sphingolipids$Value) + 7,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.Sphingolipids$Value) + 12,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "Sphingolipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
Sphingolipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/Sphingolipids_Negative_boxplot.png", plot = Sphingolipids, width = 6, height = 6, units = "in")

rm(neg.bar.Sphingolipids,anova_test,tukey_result,Sphingolipids)
```

##### Sterol Lipids 

```{r}
neg.bar.SterolLipids <- neg.bar[neg.bar$Category == "Sterol Lipids", ]

# 3 groups comparison - Surface, Pachón and Molino
# Check normality (Shapiro-Wilk test for each group)
by(neg.bar.SterolLipids$Value, neg.bar.SterolLipids$Population, shapiro.test) # MO not normally distributed

# Check homogeneity of variances (Levene's Test)
leveneTest(Value ~ Population, data = neg.bar.SterolLipids) # homogeneous

# Perform one-way ANOVA
anova_test <- aov(Value ~ Population, data = neg.bar.SterolLipids)
# Perform Tukey's HSD test - negthoc
tukey_result <- TukeyHSD(anova_test)
tukey_result <- as.data.frame(tukey_result$Population)

SterolLipids <- ggplot(neg.bar.SterolLipids, aes(x = Population, y = Value, fill = Category)) + 
  geom_boxplot(alpha = 0.7) +
  scale_fill_manual(values = pvg15[15]) +  # Specify custom colors
  geom_point(aes(x = Population), shape = 21, position = position_jitterdodge(dodge.width = 0.9)) +
  # Statistic
  geom_segment(
    aes(x = 1, xend = 2, y = max(neg.bar.SterolLipids$Value) + 2, yend = max(neg.bar.SterolLipids$Value) + 2),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[1])) + # PA vs SF
  geom_segment(
    aes(x = 1, xend = 3, y = max(neg.bar.SterolLipids$Value) + 4, yend = max(neg.bar.SterolLipids$Value) + 4),
    color = "black", size = 1,
    inherit.aes = FALSE,
    data = data.frame(p.value = tukey_result$`p adj`[2])) + # MO vs SF
  
   annotate(
    "text",
    x = 1.5,
    y = max(neg.bar.SterolLipids$Value) + 2.5,
    label = paste("p = ", format.pval(tukey_result$`p adj`[1], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # PA vs SF
  annotate(
    "text",
    x = 2,
    y = max(neg.bar.SterolLipids$Value) + 4.5,
    label = paste("p = ", format.pval(tukey_result$`p adj`[2], digits = 2)),
    hjust = 0.5, vjust = 0,
    size = 4, color = "black") + # MO vs SF
  
  theme_minimal() +
  xlab("Population") +
  ylab("Normalized abundance") +
  labs(title = "SterolLipids abundance in Negative ion mode in 2-cell stage embryos") +
  theme(text = element_text(size = 16)) +
  theme(plot.title = element_text(hjust = 0.5))  # Center the title
  
SterolLipids

#ggsave("output/vignette_run_outliersremoved_neg/median-norm/plots/Categories/all_categories/SterolLipids_Negative_boxplot.png", plot = SterolLipids, width = 6, height = 6, units = "in")

rm(neg.bar.SterolLipids,anova_test,tukey_result,SterolLipids)
```

##### Figure 2F Heat map of selected categories

```{r}
pos.heatmap <- pos.bar
pos.heatmap$Value <- log2(pos.heatmap$Value)
pos.heatmap <- pos.heatmap[pos.heatmap$Category %in% c("Sphingolipids","Carbohydrates","Nucleic acids","Organoheterocyclic compounds","Organic acids"), ]

neg.heatmap <- neg.bar
neg.heatmap$Value <- log2(neg.heatmap$Value)
neg.heatmap <- neg.heatmap[neg.heatmap$Category == "Benzenoids", ]
#neg.heatmap[neg.heatmap$Category == "Sphingolipids", "Category"] <- " Sphingolipids"

heatmap <- rbind(pos.heatmap,neg.heatmap)

heatmap <- heatmap %>%
  group_by(Category) %>%
  mutate(Z_Value = scale(Value))

heatmap$Population <- rep(c("Surface fish.1","Surface fish.2","Surface fish.3","Surface fish.4","Pachón.1","Pachón.2","Pachón.3","Pachón.4","Pachón.5","Molino.1","Molino.2","Molino.3","Molino.4","Molino.5"), 6)
heatmap$Population <- factor(heatmap$Population, levels = c("Surface fish.1","Surface fish.2","Surface fish.3","Surface fish.4","Pachón.1","Pachón.2","Pachón.3","Pachón.4","Pachón.5","Molino.1","Molino.2","Molino.3","Molino.4","Molino.5"))
heatmap$Category <- factor(heatmap$Category, levels = c("Benzenoids","Carbohydrates","Nucleic acids","Organoheterocyclic compounds","Organic acids","Sphingolipids"))

heatmap_plot <- ggplot(heatmap, aes(x = Population, y = Category, fill = Z_Value)) +
  geom_tile() +
  scale_fill_gradient2(low = "darkblue", mid = "white", high = "darkred", midpoint = 0) +  # Set color gradient
  theme_bw()+
  labs(title = "Heatmap of selected lipid super classes", x = "Population", y = "Super class", fill = "Z Value") + # Add labels
  theme(plot.title = element_text(hjust = 0.5, vjust = 0.5, size = 15),
        axis.text.y = element_text(size = 10),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.title.y = element_text(size = 15))


heatmap_plot

#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2F Heatmap of selected lipid super classes.png", plot = heatmap_plot, width = 8, height = 5, units = "in", dpi = 300)
#ggsave("output/vignette_run_outliersremoved_pos/median-norm/paper/Figure 2F Heatmap of selected lipid super classes.svg", plot = heatmap_plot, width = 8, height = 5, units = "in", dpi = 300)

rm(pos.heatmap,neg.heatmap,heatmap,heatmap_plot)
```



# Session Info
```{r}
sessionInfo()
```

